{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "%matplotlib inline\n",
    "import pandas as pd\n",
    "import itertools\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import os\n",
    "import bnpy\n",
    "import warnings\n",
    "from scipy import stats\n",
    "import random\n",
    "\n",
    "from scipy.stats import kruskal, f_oneway\n",
    "import collections\n",
    "import scikit_posthocs as sp\n",
    "\n",
    "warnings.filterwarnings('ignore')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "pth = '../data/expression/target-high-risk-nbl-mycn-na-exp-2018-11-12.tsv'\n",
    "\n",
    "exp = pd.read_csv(pth, sep='\\t', index_col=0) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "pth = '../data/archive/mycn-na-nbl-0.2-assign-2019-03-02.tsv'\n",
    "\n",
    "assign = pd.read_csv(pth, sep='\\t', index_col=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "exp.reindex(assign[assign['1'] == 1].index.values, axis=1).to_csv('target-high-risk-nbl-mycn-na-exp-2018-11-12-cluster0.tsv',\n",
    "                                                                  sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "def n1(key, zscore):\n",
    "    import uuid\n",
    "    import subprocess\n",
    "    import numpy as np\n",
    "    \n",
    "    _dir = '/tmp/%s' % str(uuid.uuid4())\n",
    "\n",
    "    os.mkdir(_dir)\n",
    "\n",
    "    cmd = [\"Rscript\",\n",
    "           \"/opt/hydra/bin/fgsea.R\",\n",
    "           \"/opt/hydra/gene-sets/Human_GOBP_AllPathways_no_GO_iea_December_01_2018_symbol.gmt\",\n",
    "           os.path.join(_dir, \"%s.rnk\" % key),\n",
    "           os.path.join(_dir, \"%s.fgsea\" % key)]\n",
    "\n",
    "    zscore = zscore.sort_values(ascending=False)\n",
    "    \n",
    "    zscore.to_csv(os.path.join(_dir, \"%s.rnk\" % key),\n",
    "                 header=None,\n",
    "                 sep='\\t')\n",
    "\n",
    "    subprocess.check_call(cmd)\n",
    "    \n",
    "    return pd.read_csv(os.path.join(_dir, \"%s.fgsea\" % key), index_col=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Run already\n"
     ]
    }
   ],
   "source": [
    "RUN = False\n",
    "\n",
    "if RUN: \n",
    "    _dir = '../data/gsea/'\n",
    "    for cluster, rows in assign.groupby('1'):\n",
    "        samps = rows.index.values\n",
    "        back = [x for x in exp.columns if x not in samps]\n",
    "    \n",
    "        for samp in samps:\n",
    "            zscore = (exp[samp] - exp[back].mean(axis=1)) / (exp[back].std(axis=1) + 0.05)\n",
    "            res = n1(samp, zscore)\n",
    "        \n",
    "            pth = os.path.join(_dir, 'sub_cohort', str(cluster), samp)\n",
    "            res.to_csv(pth)\n",
    "        \n",
    "            cback = [x for x in samps if x != samp]\n",
    "            zscore = (exp[samp] - exp[cback].mean(axis=1)) / (exp[cback].std(axis=1) + 0.05)\n",
    "            res2 = n1(samp, zscore)\n",
    "        \n",
    "            pth = os.path.join(_dir, 'sub_clust', str(cluster), samp)\n",
    "            res2.to_csv(pth)\n",
    "            \n",
    "else:\n",
    "    print(\"Run already\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TARGET-30-PALETP-01\n",
      "pathway\n",
      "COMPLEMENT ACTIVATION, CLASSICAL PATHWAY%GOBP%GO:0006958                                                                     4.787501\n",
      "COMPLEMENT ACTIVATION%GOBP%GO:0006956                                                                                        4.755103\n",
      "HUMORAL IMMUNE RESPONSE MEDIATED BY CIRCULATING IMMUNOGLOBULIN%GOBP%GO:0002455                                               4.746461\n",
      "B CELL RECEPTOR SIGNALING PATHWAY%GOBP%GO:0050853                                                                            4.611476\n",
      "PHAGOCYTOSIS, RECOGNITION%GOBP%GO:0006910                                                                                    4.383621\n",
      "IMMUNOGLOBULIN MEDIATED IMMUNE RESPONSE%GOBP%GO:0016064                                                                      4.382914\n",
      "B CELL MEDIATED IMMUNITY%GOBP%GO:0019724                                                                                     4.359972\n",
      "PROTEIN ACTIVATION CASCADE%GOBP%GO:0072376                                                                                   4.046133\n",
      "MITOCHONDRIAL ELECTRON TRANSPORT, NADH TO UBIQUINONE%GOBP%GO:0006120                                                         3.990962\n",
      "PHAGOCYTOSIS, ENGULFMENT%GOBP%GO:0006911                                                                                     3.665097\n",
      "OXIDATIVE PHOSPHORYLATION%WIKIPATHWAYS_20181110%WP623%HOMO SAPIENS                                                           3.652768\n",
      "POSITIVE REGULATION OF B CELL ACTIVATION%GOBP%GO:0050871                                                                     3.583947\n",
      "IMMUNOGLOBULIN PRODUCTION%GOBP%GO:0002377                                                                                    3.559036\n",
      "COMPLEX I BIOGENESIS%REACTOME%R-HSA-6799198.1                                                                                3.533544\n",
      "INITIAL TRIGGERING OF COMPLEMENT%REACTOME DATABASE ID RELEASE 66%166663                                                      3.357218\n",
      "NADH DEHYDROGENASE COMPLEX ASSEMBLY%GOBP%GO:0010257                                                                          3.257649\n",
      "MITOCHONDRIAL RESPIRATORY CHAIN COMPLEX I ASSEMBLY%GOBP%GO:0032981                                                           3.257649\n",
      "PRODUCTION OF MOLECULAR MEDIATOR OF IMMUNE RESPONSE%GOBP%GO:0002440                                                          3.245512\n",
      "MITOCHONDRIAL COMPLEX I ASSEMBLY MODEL OXPHOS SYSTEM%WIKIPATHWAYS_20181110%WP4324%HOMO SAPIENS                               3.201638\n",
      "PLASMA MEMBRANE INVAGINATION%GOBP%GO:0099024                                                                                 3.059698\n",
      "REGULATION OF HUMORAL IMMUNE RESPONSE%GOBP%GO:0002920                                                                        3.051709\n",
      "BIOCARTA_TCRA_PATHWAY%MSIGDB_C2%BIOCARTA_TCRA_PATHWAY                                                                        2.993317\n",
      "CD22 MEDIATED BCR REGULATION%REACTOME%R-HSA-5690714.2                                                                        2.937749\n",
      "INTERFERON ALPHA BETA SIGNALING%REACTOME%R-HSA-909733.2                                                                      2.906055\n",
      "CLASSICAL ANTIBODY-MEDIATED COMPLEMENT ACTIVATION%REACTOME%R-HSA-173623.2                                                    2.877699\n",
      "MEMBRANE INVAGINATION%GOBP%GO:0010324                                                                                        2.875905\n",
      "POSITIVE T CELL SELECTION%GOBP%GO:0043368                                                                                    2.870103\n",
      "TYPE I INTERFERON SIGNALING PATHWAY%GOBP%GO:0060337                                                                          2.860267\n",
      "CELLULAR RESPONSE TO TYPE I INTERFERON%GOBP%GO:0071357                                                                       2.860267\n",
      "RESPONSE TO TYPE I INTERFERON%GOBP%GO:0034340                                                                                2.750222\n",
      "BIOCARTA_NO2IL12_PATHWAY%MSIGDB_C2%BIOCARTA_NO2IL12_PATHWAY                                                                  2.744294\n",
      "T CELL SELECTION%GOBP%GO:0045058                                                                                             2.725299\n",
      "CREATION OF C4 AND C2 ACTIVATORS%REACTOME%R-HSA-166786.2                                                                     2.712710\n",
      "COMPLEMENT CASCADE%REACTOME DATABASE ID RELEASE 66%166658                                                                    2.705036\n",
      "REGULATION OF PROTEIN ACTIVATION CASCADE%GOBP%GO:2000257                                                                     2.699540\n",
      "REGULATION OF COMPLEMENT ACTIVATION%GOBP%GO:0030449                                                                          2.699540\n",
      "TRANSLOCATION OF ZAP-70 TO IMMUNOLOGICAL SYNAPSE%REACTOME DATABASE ID RELEASE 66%202430                                      2.635360\n",
      "BIOCARTA_THELPER_PATHWAY%MSIGDB_C2%BIOCARTA_THELPER_PATHWAY                                                                  2.560336\n",
      "POSITIVE REGULATION OF INTERLEUKIN-4 PRODUCTION%GOBP%GO:0032753                                                              2.558999\n",
      "NEGATIVE REGULATION OF LEUKOCYTE APOPTOTIC PROCESS%GOBP%GO:2000107                                                           2.553591\n",
      "ANTIGEN ACTIVATES B CELL RECEPTOR (BCR) LEADING TO GENERATION OF SECOND MESSENGERS%REACTOME DATABASE ID RELEASE 66%983695    2.550823\n",
      "BIOCARTA_TCYTOTOXIC_PATHWAY%MSIGDB_C2%BIOCARTA_TCYTOTOXIC_PATHWAY                                                            2.535522\n",
      "REGULATION OF APOPTOTIC CELL CLEARANCE%GOBP%GO:2000425                                                                       2.527910\n",
      "UNWINDING OF DNA%REACTOME%R-HSA-176974.2                                                                                     2.524899\n",
      "REGULATION OF COMPLEMENT CASCADE%REACTOME DATABASE ID RELEASE 66%977606                                                      2.512088\n",
      "B CELL PROLIFERATION%GOBP%GO:0042100                                                                                         2.507712\n",
      "FCGR ACTIVATION%REACTOME%R-HSA-2029481.1                                                                                     2.472586\n",
      "IMMUNOLOGICAL SYNAPSE FORMATION%GOBP%GO:0001771                                                                              2.467722\n",
      "POSITIVE REGULATION OF PHAGOCYTOSIS%GOBP%GO:0050766                                                                          2.446452\n",
      "THYMIC T CELL SELECTION%GOBP%GO:0045061                                                                                      2.439416\n",
      "Name: NES, dtype: float64\n",
      "\n",
      "\n",
      "pathway\n",
      "PEPTIDE CROSS-LINKING%GOBP%GO:0018149                                   2.817382\n",
      "MITOCHONDRIAL ATP SYNTHESIS COUPLED PROTON TRANSPORT%GOBP%GO:0042776    2.537321\n",
      "FORMATION OF ATP BY CHEMIOSMOTIC COUPLING%REACTOME%R-HSA-163210.2       2.366126\n",
      "Name: NES, dtype: float64\n",
      "\n",
      "\n",
      "\n",
      "TARGET-30-PALIIN-01\n",
      "pathway\n",
      "HALLMARK_E2F_TARGETS%MSIGDB_C2%HALLMARK_E2F_TARGETS                                                                          2.754293\n",
      "B CELL RECEPTOR SIGNALING PATHWAY%GOBP%GO:0050853                                                                            2.738211\n",
      "LYMPHOCYTE ACTIVATION%GOBP%GO:0046649                                                                                        2.718012\n",
      "LYMPHOCYTE DIFFERENTIATION%GOBP%GO:0030098                                                                                   2.670934\n",
      "POSITIVE T CELL SELECTION%GOBP%GO:0043368                                                                                    2.635263\n",
      "DNA-DEPENDENT DNA REPLICATION%GOBP%GO:0006261                                                                                2.616474\n",
      "DNA REPLICATION%WIKIPATHWAYS_20181110%WP466%HOMO SAPIENS                                                                     2.605107\n",
      "B CELL ACTIVATION%GOBP%GO:0042113                                                                                            2.569891\n",
      "B CELL PROLIFERATION%GOBP%GO:0042100                                                                                         2.539009\n",
      "T CELL SELECTION%GOBP%GO:0045058                                                                                             2.530246\n",
      "LYMPHOCYTE PROLIFERATION%GOBP%GO:0046651                                                                                     2.528744\n",
      "MONONUCLEAR CELL PROLIFERATION%GOBP%GO:0032943                                                                               2.499018\n",
      "DNA REPLICATION%GOBP%GO:0006260                                                                                              2.473973\n",
      "RETINOBLASTOMA GENE IN CANCER%WIKIPATHWAYS_20181110%WP2446%HOMO SAPIENS                                                      2.465108\n",
      "DNA REPLICATION%REACTOME DATABASE ID RELEASE 66%69306                                                                        2.459977\n",
      "HALLMARK_G2M_CHECKPOINT%MSIGDB_C2%HALLMARK_G2M_CHECKPOINT                                                                    2.455585\n",
      "ANTIGEN RECEPTOR-MEDIATED SIGNALING PATHWAY%GOBP%GO:0050851                                                                  2.437699\n",
      "DNA STRAND ELONGATION%REACTOME DATABASE ID RELEASE 66%69190                                                                  2.433159\n",
      "POSITIVE REGULATION OF LYMPHOCYTE ACTIVATION%GOBP%GO:0051251                                                                 2.421431\n",
      "T CELL ACTIVATION%GOBP%GO:0042110                                                                                            2.416912\n",
      "POSITIVE REGULATION OF B CELL ACTIVATION%GOBP%GO:0050871                                                                     2.409220\n",
      "NUCLEAR DNA REPLICATION%GOBP%GO:0033260                                                                                      2.395011\n",
      "POSITIVE REGULATION OF LEUKOCYTE ACTIVATION%GOBP%GO:0002696                                                                  2.390092\n",
      "SYNTHESIS OF DNA%REACTOME DATABASE ID RELEASE 66%69239                                                                       2.375891\n",
      "B CELL DIFFERENTIATION%GOBP%GO:0030183                                                                                       2.375538\n",
      "POSITIVE REGULATION OF CELL ACTIVATION%GOBP%GO:0050867                                                                       2.373333\n",
      "S PHASE%REACTOME DATABASE ID RELEASE 66%69242                                                                                2.368361\n",
      "ACTIVATION OF ATR IN RESPONSE TO REPLICATION STRESS%REACTOME%R-HSA-176187.2                                                  2.353602\n",
      "LEUKOCYTE PROLIFERATION%GOBP%GO:0070661                                                                                      2.352003\n",
      "HALLMARK_ALLOGRAFT_REJECTION%MSIGDB_C2%HALLMARK_ALLOGRAFT_REJECTION                                                          2.351731\n",
      "ACTIVATION OF THE PRE-REPLICATIVE COMPLEX%REACTOME%R-HSA-68962.3                                                             2.344778\n",
      "IMMUNOLOGICAL SYNAPSE FORMATION%GOBP%GO:0001771                                                                              2.335594\n",
      "REGULATION OF B CELL ACTIVATION%GOBP%GO:0050864                                                                              2.329117\n",
      "T CELL DIFFERENTIATION%GOBP%GO:0030217                                                                                       2.324072\n",
      "CELL CYCLE DNA REPLICATION%GOBP%GO:0044786                                                                                   2.319542\n",
      "COMPLEMENT ACTIVATION, CLASSICAL PATHWAY%GOBP%GO:0006958                                                                     2.319147\n",
      "PHAGOCYTOSIS, RECOGNITION%GOBP%GO:0006910                                                                                    2.307139\n",
      "REGULATION OF LYMPHOCYTE ACTIVATION%GOBP%GO:0051249                                                                          2.295239\n",
      "ANTIGEN ACTIVATES B CELL RECEPTOR (BCR) LEADING TO GENERATION OF SECOND MESSENGERS%REACTOME DATABASE ID RELEASE 66%983695    2.289969\n",
      "HALLMARK_INTERFERON_ALPHA_RESPONSE%MSIGDB_C2%HALLMARK_INTERFERON_ALPHA_RESPONSE                                              2.280297\n",
      "HALLMARK_INTERFERON_GAMMA_RESPONSE%MSIGDB_C2%HALLMARK_INTERFERON_GAMMA_RESPONSE                                              2.272374\n",
      "BIOCARTA_THELPER_PATHWAY%MSIGDB_C2%BIOCARTA_THELPER_PATHWAY                                                                  2.265961\n",
      "G1 S-SPECIFIC TRANSCRIPTION%REACTOME DATABASE ID RELEASE 66%69205                                                            2.247449\n",
      "ACTIVATION OF E2F1 TARGET GENES AT G1 S%REACTOME%R-HSA-539107.4                                                              2.247449\n",
      "PCNA-DEPENDENT LONG PATCH BASE EXCISION REPAIR%REACTOME DATABASE ID RELEASE 66%5651801                                       2.246524\n",
      "LEUKOCYTE DIFFERENTIATION%GOBP%GO:0002521                                                                                    2.242119\n",
      "DNA REPLICATION PRE-INITIATION%REACTOME DATABASE ID RELEASE 66%69002                                                         2.235916\n",
      "ADAPTIVE IMMUNE RESPONSE%GOBP%GO:0002250                                                                                     2.231344\n",
      "BIOCARTA_NO2IL12_PATHWAY%MSIGDB_C2%BIOCARTA_NO2IL12_PATHWAY                                                                  2.228138\n",
      "NATURAL KILLER CELL ACTIVATION%GOBP%GO:0030101                                                                               2.226507\n",
      "Name: NES, dtype: float64\n",
      "\n",
      "\n",
      "Series([], Name: NES, dtype: float64)\n",
      "\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TARGET-30-PALTEG-01\n",
      "pathway\n",
      "EUKARYOTIC TRANSLATION ELONGATION%REACTOME DATABASE ID RELEASE 66%156842                                                                        7.153070\n",
      "PEPTIDE CHAIN ELONGATION%REACTOME DATABASE ID RELEASE 66%156902                                                                                 7.103036\n",
      "FORMATION OF A POOL OF FREE 40S SUBUNITS%REACTOME DATABASE ID RELEASE 66%72689                                                                  7.029831\n",
      "VIRAL MRNA TRANSLATION%REACTOME%R-HSA-192823.3                                                                                                  6.943775\n",
      "EUKARYOTIC TRANSLATION TERMINATION%REACTOME%R-HSA-72764.4                                                                                       6.875884\n",
      "CYTOPLASMIC RIBOSOMAL PROTEINS%WIKIPATHWAYS_20181110%WP477%HOMO SAPIENS                                                                         6.803590\n",
      "SELENOCYSTEINE SYNTHESIS%REACTOME%R-HSA-2408557.2                                                                                               6.761527\n",
      "L13A-MEDIATED TRANSLATIONAL SILENCING OF CERULOPLASMIN EXPRESSION%REACTOME DATABASE ID RELEASE 66%156827                                        6.721698\n",
      "NONSENSE MEDIATED DECAY (NMD) INDEPENDENT OF THE EXON JUNCTION COMPLEX (EJC)%REACTOME DATABASE ID RELEASE 66%975956                             6.708533\n",
      "GTP HYDROLYSIS AND JOINING OF THE 60S RIBOSOMAL SUBUNIT%REACTOME DATABASE ID RELEASE 66%72706                                                   6.670461\n",
      "EUKARYOTIC TRANSLATION INITIATION%REACTOME DATABASE ID RELEASE 66%72613                                                                         6.564757\n",
      "CAP-DEPENDENT TRANSLATION INITIATION%REACTOME DATABASE ID RELEASE 66%72737                                                                      6.564757\n",
      "SRP-DEPENDENT COTRANSLATIONAL PROTEIN TARGETING TO MEMBRANE%GOBP%GO:0006614                                                                     6.512765\n",
      "SRP-DEPENDENT COTRANSLATIONAL PROTEIN TARGETING TO MEMBRANE%REACTOME%R-HSA-1799339.1                                                            6.237409\n",
      "COTRANSLATIONAL PROTEIN TARGETING TO MEMBRANE%GOBP%GO:0006613                                                                                   6.203670\n",
      "SELENOAMINO ACID METABOLISM%REACTOME%R-HSA-2408522.2                                                                                            6.170056\n",
      "PROTEIN TARGETING TO ER%GOBP%GO:0045047                                                                                                         6.121704\n",
      "TRANSLATIONAL INITIATION%GOBP%GO:0006413                                                                                                        6.117930\n",
      "INFLUENZA VIRAL RNA TRANSCRIPTION AND REPLICATION%REACTOME%R-HSA-168273.3                                                                       5.895582\n",
      "ESTABLISHMENT OF PROTEIN LOCALIZATION TO ENDOPLASMIC RETICULUM%GOBP%GO:0072599                                                                  5.868131\n",
      "NONSENSE-MEDIATED DECAY (NMD)%REACTOME DATABASE ID RELEASE 66%927802                                                                            5.803571\n",
      "NONSENSE MEDIATED DECAY (NMD) ENHANCED BY THE EXON JUNCTION COMPLEX (EJC)%REACTOME%R-HSA-975957.1                                               5.803571\n",
      "INFLUENZA LIFE CYCLE%REACTOME DATABASE ID RELEASE 66%168255                                                                                     5.668187\n",
      "NUCLEAR-TRANSCRIBED MRNA CATABOLIC PROCESS, NONSENSE-MEDIATED DECAY%GOBP%GO:0000184                                                             5.656832\n",
      "FORMATION OF THE TERNARY COMPLEX, AND SUBSEQUENTLY, THE 43S COMPLEX%REACTOME%R-HSA-72695.2                                                      5.460292\n",
      "INFLUENZA INFECTION%REACTOME%R-HSA-168254.2                                                                                                     5.449216\n",
      "REGULATION OF EXPRESSION OF SLITS AND ROBOS%REACTOME DATABASE ID RELEASE 66%9010553                                                             5.297644\n",
      "ACTIVATION OF THE MRNA UPON BINDING OF THE CAP-BINDING COMPLEX AND EIFS, AND SUBSEQUENT BINDING TO 43S%REACTOME DATABASE ID RELEASE 66%72662    5.158624\n",
      "TRANSLATION INITIATION COMPLEX FORMATION%REACTOME DATABASE ID RELEASE 66%72649                                                                  5.126468\n",
      "PROTEIN LOCALIZATION TO ENDOPLASMIC RETICULUM%GOBP%GO:0070972                                                                                   5.114966\n",
      "RIBOSOMAL SCANNING AND START CODON RECOGNITION%REACTOME DATABASE ID RELEASE 66%72702                                                            5.065665\n",
      "HUMORAL IMMUNE RESPONSE%GOBP%GO:0006959                                                                                                         4.927261\n",
      "IMMUNOREGULATORY INTERACTIONS BETWEEN A LYMPHOID AND A NON-LYMPHOID CELL%REACTOME DATABASE ID RELEASE 66%198933                                 4.826941\n",
      "PROTEIN ACTIVATION CASCADE%GOBP%GO:0072376                                                                                                      4.812405\n",
      "PROTEIN TARGETING TO MEMBRANE%GOBP%GO:0006612                                                                                                   4.674580\n",
      "HUMORAL IMMUNE RESPONSE MEDIATED BY CIRCULATING IMMUNOGLOBULIN%GOBP%GO:0002455                                                                  4.648518\n",
      "CYTOPLASMIC TRANSLATION%GOBP%GO:0002181                                                                                                         4.457447\n",
      "COMPLEMENT ACTIVATION%GOBP%GO:0006956                                                                                                           4.425995\n",
      "COMPLEMENT ACTIVATION, CLASSICAL PATHWAY%GOBP%GO:0006958                                                                                        4.394993\n",
      "B CELL MEDIATED IMMUNITY%GOBP%GO:0019724                                                                                                        4.352096\n",
      "B CELL RECEPTOR SIGNALING PATHWAY%GOBP%GO:0050853                                                                                               4.346505\n",
      "IMMUNOGLOBULIN MEDIATED IMMUNE RESPONSE%GOBP%GO:0016064                                                                                         4.246419\n",
      "REGULATION OF HUMORAL IMMUNE RESPONSE%GOBP%GO:0002920                                                                                           4.102420\n",
      "PHAGOCYTOSIS, RECOGNITION%GOBP%GO:0006910                                                                                                       4.098252\n",
      "DEFENSE RESPONSE TO BACTERIUM%GOBP%GO:0042742                                                                                                   4.049236\n",
      "ADAPTIVE IMMUNE RESPONSE%GOBP%GO:0002250                                                                                                        3.920162\n",
      "ADAPTIVE IMMUNE RESPONSE BASED ON SOMATIC RECOMBINATION OF IMMUNE RECEPTORS BUILT FROM IMMUNOGLOBULIN SUPERFAMILY DOMAINS%GOBP%GO:0002460       3.891937\n",
      "POSITIVE REGULATION OF MONONUCLEAR CELL PROLIFERATION%GOBP%GO:0032946                                                                           3.805869\n",
      "POSITIVE REGULATION OF LEUKOCYTE PROLIFERATION%GOBP%GO:0070665                                                                                  3.773306\n",
      "REGULATION OF MONONUCLEAR CELL PROLIFERATION%GOBP%GO:0032944                                                                                    3.744939\n",
      "Name: NES, dtype: float64\n",
      "\n",
      "\n",
      "pathway\n",
      "BLOOD CLOTTING CASCADE%WIKIPATHWAYS_20181110%WP272%HOMO SAPIENS                                                        2.943072\n",
      "BIOCARTA_AMI_PATHWAY%MSIGDB_C2%BIOCARTA_AMI_PATHWAY                                                                    2.633584\n",
      "RENAL SYSTEM PROCESS INVOLVED IN REGULATION OF SYSTEMIC ARTERIAL BLOOD PRESSURE%GOBP%GO:0003071                        2.630728\n",
      "ERYTHROCYTES TAKE UP OXYGEN AND RELEASE CARBON DIOXIDE%REACTOME%R-HSA-1247673.1                                        2.595476\n",
      "PHENOL-CONTAINING COMPOUND CATABOLIC PROCESS%GOBP%GO:0019336                                                           2.594803\n",
      "FOLDING OF ACTIN BY CCT TRIC%REACTOME DATABASE ID RELEASE 66%390450                                                    2.594025\n",
      "BLOOD COAGULATION, INTRINSIC PATHWAY%GOBP%GO:0007597                                                                   2.589220\n",
      "MITOCHONDRIAL ELECTRON TRANSPORT, UBIQUINOL TO CYTOCHROME C%GOBP%GO:0006122                                            2.571783\n",
      "PLASMINOGEN ACTIVATING CASCADE%PANTHER PATHWAY%P00050                                                                  2.552789\n",
      "ESTABLISHMENT OF PROTEIN LOCALIZATION TO MITOCHONDRIAL MEMBRANE%GOBP%GO:0090151                                        2.537470\n",
      "BIOGENIC AMINE SYNTHESIS%WIKIPATHWAYS_20181110%WP550%HOMO SAPIENS                                                      2.516486\n",
      "POSITIVE REGULATION OF INTERLEUKIN-8 SECRETION%GOBP%GO:2000484                                                         2.516411\n",
      "BIOCARTA_CTL_PATHWAY%MSIGDB_C2%BIOCARTA_CTL_PATHWAY                                                                    2.472221\n",
      "MONOCYTE DIFFERENTIATION%GOBP%GO:0030224                                                                               2.468859\n",
      "MONONUCLEAR CELL DIFFERENTIATION%GOBP%GO:1903131                                                                       2.468859\n",
      "REGULATION OF DENDRITIC CELL CHEMOTAXIS%GOBP%GO:2000508                                                                2.465342\n",
      "CATECHOL-CONTAINING COMPOUND CATABOLIC PROCESS%GOBP%GO:0019614                                                         2.443711\n",
      "CATECHOLAMINE CATABOLIC PROCESS%GOBP%GO:0042424                                                                        2.443711\n",
      "BIOCARTA_CLASSIC_PATHWAY%MSIGDB_C2%BIOCARTA_CLASSIC_PATHWAY                                                            2.427177\n",
      "BIOCARTA_IL17_PATHWAY%MSIGDB_C2%BIOCARTA_IL17_PATHWAY                                                                  2.404190\n",
      "MACROPINOCYTOSIS%GOBP%GO:0044351                                                                                       2.372625\n",
      "FIBRINOLYSIS%GOBP%GO:0042730                                                                                           2.371728\n",
      "BIOCARTA_INTRINSIC_PATHWAY%MSIGDB_C2%BIOCARTA_INTRINSIC_PATHWAY                                                        2.370346\n",
      "REGULATION OF RENAL SYSTEM PROCESS%GOBP%GO:0098801                                                                     2.355570\n",
      "DEFECTIVE B4GALT1 CAUSES B4GALT1-CDG (CDG-2D)%REACTOME DATABASE ID RELEASE 66%3656244                                  2.243753\n",
      "DEFECTIVE CHST6 CAUSES MCDC1%REACTOME DATABASE ID RELEASE 66%3656225                                                   2.243753\n",
      "DEFECTIVE ST3GAL3 CAUSES MCT12 AND EIEE15%REACTOME DATABASE ID RELEASE 66%3656243                                      2.243753\n",
      "MESENCHYME MIGRATION%GOBP%GO:0090131                                                                                   1.700442\n",
      "ENDONUCLEOLYTIC CLEAVAGE TO GENERATE MATURE 3'-END OF SSU-RRNA FROM (SSU-RRNA, 5.8S RRNA, LSU-RRNA)%GOBP%GO:0000461    1.693822\n",
      "NEGATIVE REGULATION OF CALCIUM-TRANSPORTING ATPASE ACTIVITY%GOBP%GO:1901895                                            1.690512\n",
      "Name: NES, dtype: float64\n",
      "\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TARGET-30-PANNMS-01\n",
      "pathway\n",
      "HALLMARK_E2F_TARGETS%MSIGDB_C2%HALLMARK_E2F_TARGETS                                                                         4.096941\n",
      "HALLMARK_MYC_TARGETS_V1%MSIGDB_C2%HALLMARK_MYC_TARGETS_V1                                                                   3.596211\n",
      "HALLMARK_G2M_CHECKPOINT%MSIGDB_C2%HALLMARK_G2M_CHECKPOINT                                                                   3.446443\n",
      "RETINOBLASTOMA GENE IN CANCER%WIKIPATHWAYS_20181110%WP2446%HOMO SAPIENS                                                     3.389049\n",
      "RESOLUTION OF SISTER CHROMATID COHESION%REACTOME DATABASE ID RELEASE 66%2500257                                             3.355393\n",
      "CELL CYCLE, MITOTIC%REACTOME%R-HSA-69278.3                                                                                  3.271655\n",
      "MITOTIC PROMETAPHASE%REACTOME DATABASE ID RELEASE 66%68877                                                                  3.261985\n",
      "GTP HYDROLYSIS AND JOINING OF THE 60S RIBOSOMAL SUBUNIT%REACTOME DATABASE ID RELEASE 66%72706                               3.257413\n",
      "L13A-MEDIATED TRANSLATIONAL SILENCING OF CERULOPLASMIN EXPRESSION%REACTOME DATABASE ID RELEASE 66%156827                    3.243573\n",
      "S PHASE%REACTOME DATABASE ID RELEASE 66%69242                                                                               3.215131\n",
      "SEPARATION OF SISTER CHROMATIDS%REACTOME%R-HSA-2467813.1                                                                    3.214326\n",
      "MITOTIC ANAPHASE%REACTOME%R-HSA-68882.2                                                                                     3.212939\n",
      "EUKARYOTIC TRANSLATION INITIATION%REACTOME DATABASE ID RELEASE 66%72613                                                     3.200704\n",
      "CAP-DEPENDENT TRANSLATION INITIATION%REACTOME DATABASE ID RELEASE 66%72737                                                  3.200704\n",
      "FORMATION OF A POOL OF FREE 40S SUBUNITS%REACTOME DATABASE ID RELEASE 66%72689                                              3.200082\n",
      "MITOTIC METAPHASE AND ANAPHASE%REACTOME DATABASE ID RELEASE 66%2555396                                                      3.199140\n",
      "DNA REPLICATION%REACTOME DATABASE ID RELEASE 66%69306                                                                       3.191389\n",
      "G1 S TRANSITION%REACTOME DATABASE ID RELEASE 66%69206                                                                       3.187883\n",
      "MITOTIC G1-G1 S PHASES%REACTOME%R-HSA-453279.3                                                                              3.181555\n",
      "CELL CYCLE CHECKPOINTS%REACTOME%R-HSA-69620.1                                                                               3.179489\n",
      "SRP-DEPENDENT COTRANSLATIONAL PROTEIN TARGETING TO MEMBRANE%REACTOME%R-HSA-1799339.1                                        3.167699\n",
      "EUKARYOTIC TRANSLATION ELONGATION%REACTOME DATABASE ID RELEASE 66%156842                                                    3.153922\n",
      "B CELL RECEPTOR SIGNALING PATHWAY%GOBP%GO:0050853                                                                           3.149273\n",
      "TRANSLATIONAL INITIATION%GOBP%GO:0006413                                                                                    3.142821\n",
      "DNA REPLICATION PRE-INITIATION%REACTOME DATABASE ID RELEASE 66%69002                                                        3.140589\n",
      "AMPLIFICATION OF SIGNAL FROM UNATTACHED KINETOCHORES VIA A MAD2 INHIBITORY SIGNAL%REACTOME DATABASE ID RELEASE 66%141444    3.135233\n",
      "AMPLIFICATION OF SIGNAL FROM THE KINETOCHORES%REACTOME%R-HSA-141424.2                                                       3.135233\n",
      "PROTEIN TARGETING TO ER%GOBP%GO:0045047                                                                                     3.099881\n",
      "SYNTHESIS OF DNA%REACTOME DATABASE ID RELEASE 66%69239                                                                      3.082828\n",
      "SRP-DEPENDENT COTRANSLATIONAL PROTEIN TARGETING TO MEMBRANE%GOBP%GO:0006614                                                 3.069173\n",
      "PEPTIDE CHAIN ELONGATION%REACTOME DATABASE ID RELEASE 66%156902                                                             3.067749\n",
      "TRANSLATION%REACTOME%R-HSA-72766.3                                                                                          3.061467\n",
      "REGULATION OF EXPRESSION OF SLITS AND ROBOS%REACTOME DATABASE ID RELEASE 66%9010553                                         3.055512\n",
      "ACTIVATION OF THE PRE-REPLICATIVE COMPLEX%REACTOME%R-HSA-68962.3                                                            3.039069\n",
      "CYTOPLASMIC RIBOSOMAL PROTEINS%WIKIPATHWAYS_20181110%WP477%HOMO SAPIENS                                                     3.020497\n",
      "COMPLEMENT ACTIVATION, CLASSICAL PATHWAY%GOBP%GO:0006958                                                                    3.019256\n",
      "B CELL MEDIATED IMMUNITY%GOBP%GO:0019724                                                                                    3.015964\n",
      "RHO GTPASES ACTIVATE FORMINS%REACTOME%R-HSA-5663220.1                                                                       3.002459\n",
      "EUKARYOTIC TRANSLATION TERMINATION%REACTOME%R-HSA-72764.4                                                                   2.995851\n",
      "ESTABLISHMENT OF PROTEIN LOCALIZATION TO ENDOPLASMIC RETICULUM%GOBP%GO:0072599                                              2.981594\n",
      "COTRANSLATIONAL PROTEIN TARGETING TO MEMBRANE%GOBP%GO:0006613                                                               2.980707\n",
      "MITOTIC G2-G2 M PHASES%REACTOME%R-HSA-453274.2                                                                              2.978794\n",
      "SIGNALING BY ROBO RECEPTORS%REACTOME%R-HSA-376176.4                                                                         2.969337\n",
      "IMMUNOGLOBULIN MEDIATED IMMUNE RESPONSE%GOBP%GO:0016064                                                                     2.954196\n",
      "G2 M TRANSITION%REACTOME%R-HSA-69275.5                                                                                      2.950657\n",
      "NONSENSE MEDIATED DECAY (NMD) INDEPENDENT OF THE EXON JUNCTION COMPLEX (EJC)%REACTOME DATABASE ID RELEASE 66%975956         2.950542\n",
      "SELENOCYSTEINE SYNTHESIS%REACTOME%R-HSA-2408557.2                                                                           2.940176\n",
      "INFECTIOUS DISEASE%REACTOME DATABASE ID RELEASE 66%5663205                                                                  2.938228\n",
      "MITOTIC SPINDLE CHECKPOINT%REACTOME%R-HSA-69618.2                                                                           2.930990\n",
      "VIRAL MRNA TRANSLATION%REACTOME%R-HSA-192823.3                                                                              2.927108\n",
      "Name: NES, dtype: float64\n",
      "\n",
      "\n",
      "pathway\n",
      "L13A-MEDIATED TRANSLATIONAL SILENCING OF CERULOPLASMIN EXPRESSION%REACTOME DATABASE ID RELEASE 66%156827                                        4.761352\n",
      "GTP HYDROLYSIS AND JOINING OF THE 60S RIBOSOMAL SUBUNIT%REACTOME DATABASE ID RELEASE 66%72706                                                   4.739594\n",
      "HALLMARK_MYC_TARGETS_V1%MSIGDB_C2%HALLMARK_MYC_TARGETS_V1                                                                                       4.685319\n",
      "EUKARYOTIC TRANSLATION INITIATION%REACTOME DATABASE ID RELEASE 66%72613                                                                         4.631860\n",
      "CAP-DEPENDENT TRANSLATION INITIATION%REACTOME DATABASE ID RELEASE 66%72737                                                                      4.631860\n",
      "FORMATION OF A POOL OF FREE 40S SUBUNITS%REACTOME DATABASE ID RELEASE 66%72689                                                                  4.622581\n",
      "TRANSLATIONAL INITIATION%GOBP%GO:0006413                                                                                                        4.510059\n",
      "SRP-DEPENDENT COTRANSLATIONAL PROTEIN TARGETING TO MEMBRANE%REACTOME%R-HSA-1799339.1                                                            4.456561\n",
      "EUKARYOTIC TRANSLATION ELONGATION%REACTOME DATABASE ID RELEASE 66%156842                                                                        4.431001\n",
      "PEPTIDE CHAIN ELONGATION%REACTOME DATABASE ID RELEASE 66%156902                                                                                 4.375338\n",
      "SRP-DEPENDENT COTRANSLATIONAL PROTEIN TARGETING TO MEMBRANE%GOBP%GO:0006614                                                                     4.333138\n",
      "CYTOPLASMIC RIBOSOMAL PROTEINS%WIKIPATHWAYS_20181110%WP477%HOMO SAPIENS                                                                         4.330478\n",
      "PROTEIN TARGETING TO ER%GOBP%GO:0045047                                                                                                         4.286878\n",
      "COTRANSLATIONAL PROTEIN TARGETING TO MEMBRANE%GOBP%GO:0006613                                                                                   4.269862\n",
      "VIRAL MRNA TRANSLATION%REACTOME%R-HSA-192823.3                                                                                                  4.268118\n",
      "TRANSLATION%REACTOME%R-HSA-72766.3                                                                                                              4.208194\n",
      "EUKARYOTIC TRANSLATION TERMINATION%REACTOME%R-HSA-72764.4                                                                                       4.201819\n",
      "SELENOCYSTEINE SYNTHESIS%REACTOME%R-HSA-2408557.2                                                                                               4.156835\n",
      "NONSENSE MEDIATED DECAY (NMD) INDEPENDENT OF THE EXON JUNCTION COMPLEX (EJC)%REACTOME DATABASE ID RELEASE 66%975956                             4.129868\n",
      "ESTABLISHMENT OF PROTEIN LOCALIZATION TO ENDOPLASMIC RETICULUM%GOBP%GO:0072599                                                                  4.129001\n",
      "TRANSLATION INITIATION COMPLEX FORMATION%REACTOME DATABASE ID RELEASE 66%72649                                                                  4.127526\n",
      "ACTIVATION OF THE MRNA UPON BINDING OF THE CAP-BINDING COMPLEX AND EIFS, AND SUBSEQUENT BINDING TO 43S%REACTOME DATABASE ID RELEASE 66%72662    4.078122\n",
      "RIBOSOMAL SCANNING AND START CODON RECOGNITION%REACTOME DATABASE ID RELEASE 66%72702                                                            4.065247\n",
      "FORMATION OF THE TERNARY COMPLEX, AND SUBSEQUENTLY, THE 43S COMPLEX%REACTOME%R-HSA-72695.2                                                      4.044408\n",
      "REGULATION OF EXPRESSION OF SLITS AND ROBOS%REACTOME DATABASE ID RELEASE 66%9010553                                                             3.961995\n",
      "SIGNALING BY ROBO RECEPTORS%REACTOME%R-HSA-376176.4                                                                                             3.917222\n",
      "SELENOAMINO ACID METABOLISM%REACTOME%R-HSA-2408522.2                                                                                            3.916866\n",
      "PROTEIN LOCALIZATION TO ENDOPLASMIC RETICULUM%GOBP%GO:0070972                                                                                   3.853908\n",
      "INFLUENZA LIFE CYCLE%REACTOME DATABASE ID RELEASE 66%168255                                                                                     3.840594\n",
      "PROTEIN TARGETING TO MEMBRANE%GOBP%GO:0006612                                                                                                   3.838905\n",
      "TRANSLATION%GOBP%GO:0006412                                                                                                                     3.820570\n",
      "INFLUENZA VIRAL RNA TRANSCRIPTION AND REPLICATION%REACTOME%R-HSA-168273.3                                                                       3.816165\n",
      "NONSENSE-MEDIATED DECAY (NMD)%REACTOME DATABASE ID RELEASE 66%927802                                                                            3.794783\n",
      "NONSENSE MEDIATED DECAY (NMD) ENHANCED BY THE EXON JUNCTION COMPLEX (EJC)%REACTOME%R-HSA-975957.1                                               3.794783\n",
      "INFLUENZA INFECTION%REACTOME%R-HSA-168254.2                                                                                                     3.714589\n",
      "HALLMARK_OXIDATIVE_PHOSPHORYLATION%MSIGDB_C2%HALLMARK_OXIDATIVE_PHOSPHORYLATION                                                                 3.675182\n",
      "INFECTIOUS DISEASE%REACTOME DATABASE ID RELEASE 66%5663205                                                                                      3.666353\n",
      "PEPTIDE BIOSYNTHETIC PROCESS%GOBP%GO:0043043                                                                                                    3.660980\n",
      "NUCLEAR-TRANSCRIBED MRNA CATABOLIC PROCESS, NONSENSE-MEDIATED DECAY%GOBP%GO:0000184                                                             3.627171\n",
      "CYTOPLASMIC TRANSLATION%GOBP%GO:0002181                                                                                                         3.591008\n",
      "HALLMARK_E2F_TARGETS%MSIGDB_C2%HALLMARK_E2F_TARGETS                                                                                             3.405058\n",
      "AMIDE BIOSYNTHETIC PROCESS%GOBP%GO:0043604                                                                                                      3.393445\n",
      "PEPTIDE METABOLIC PROCESS%GOBP%GO:0006518                                                                                                       3.382279\n",
      "PROTEIN TARGETING%GOBP%GO:0006605                                                                                                               3.292989\n",
      "DNA REPLICATION PRE-INITIATION%REACTOME DATABASE ID RELEASE 66%69002                                                                            3.257135\n",
      "PREFOLDIN MEDIATED TRANSFER OF SUBSTRATE TO CCT TRIC%REACTOME DATABASE ID RELEASE 66%389957                                                     3.214384\n",
      "AMINO ACID AND DERIVATIVE METABOLISM%REACTOME%R-HSA-71291.4                                                                                     3.196319\n",
      "MAJOR PATHWAY OF RRNA PROCESSING IN THE NUCLEOLUS AND CYTOSOL%REACTOME DATABASE ID RELEASE 66%6791226                                           3.173109\n",
      "COOPERATION OF PREFOLDIN AND TRIC CCT IN ACTIN AND TUBULIN FOLDING%REACTOME%R-HSA-389958.2                                                      3.128089\n",
      "ESTABLISHMENT OF PROTEIN LOCALIZATION TO MEMBRANE%GOBP%GO:0090150                                                                               3.108890\n",
      "Name: NES, dtype: float64\n",
      "\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TARGET-30-PANRHJ-01\n",
      "Series([], Name: NES, dtype: float64)\n",
      "\n",
      "\n",
      "pathway\n",
      "REGULATION OF CYTOCHROME-C OXIDASE ACTIVITY%GOBP%GO:1904959             1.682677\n",
      "POSITIVE REGULATION OF CYTOCHROME-C OXIDASE ACTIVITY%GOBP%GO:1904960    1.682677\n",
      "POSITIVE REGULATION OF ELECTRON TRANSFER ACTIVITY%GOBP%GO:1904734       1.682677\n",
      "Name: NES, dtype: float64\n",
      "\n",
      "\n",
      "\n",
      "TARGET-30-PANZVU-01\n",
      "pathway\n",
      "B CELL RECEPTOR SIGNALING PATHWAY%GOBP%GO:0050853                                                                               2.960503\n",
      "ANTIGEN RECEPTOR-MEDIATED SIGNALING PATHWAY%GOBP%GO:0050851                                                                     2.742906\n",
      "ANTIGEN ACTIVATES B CELL RECEPTOR (BCR) LEADING TO GENERATION OF SECOND MESSENGERS%REACTOME DATABASE ID RELEASE 66%983695       2.636827\n",
      "B CELL ACTIVATION%GOBP%GO:0042113                                                                                               2.587924\n",
      "PID_TCR_PATHWAY%MSIGDB_C2%PID_TCR_PATHWAY                                                                                       2.523629\n",
      "TCR SIGNALING IN NAIVE CD4+ T CELLS%PATHWAY INTERACTION DATABASE NCI-NATURE CURATED DATA%TCR SIGNALING IN NAIVE CD4+ T CELLS    2.501767\n",
      "FC-EPSILON RECEPTOR SIGNALING PATHWAY%GOBP%GO:0038095                                                                           2.496528\n",
      "FCERI MEDIATED CA+2 MOBILIZATION%REACTOME%R-HSA-2871809.1                                                                       2.480677\n",
      "ROLE OF LAT2 NTAL LAB ON CALCIUM MOBILIZATION%REACTOME DATABASE ID RELEASE 66%2730905                                           2.462942\n",
      "BCR SIGNALING PATHWAY%PATHWAY INTERACTION DATABASE NCI-NATURE CURATED DATA%BCR SIGNALING PATHWAY                                2.455366\n",
      "PID_BCR_5PATHWAY%MSIGDB_C2%PID_BCR_5PATHWAY                                                                                     2.437981\n",
      "TCR SIGNALING IN NAIVE CD8+ T CELLS%PATHWAY INTERACTION DATABASE NCI-NATURE CURATED DATA%TCR SIGNALING IN NAIVE CD8+ T CELLS    2.435052\n",
      "PID_CD8_TCR_PATHWAY%MSIGDB_C2%PID_CD8_TCR_PATHWAY                                                                               2.435052\n",
      "COMPLEMENT ACTIVATION, CLASSICAL PATHWAY%GOBP%GO:0006958                                                                        2.421125\n",
      "POSITIVE T CELL SELECTION%GOBP%GO:0043368                                                                                       2.411958\n",
      "T CELL RECEPTOR SIGNALING PATHWAY%GOBP%GO:0050852                                                                               2.411186\n",
      "LYMPHOCYTE ACTIVATION%GOBP%GO:0046649                                                                                           2.382554\n",
      "B CELL RECEPTOR SIGNALING PATHWAY%WIKIPATHWAYS_20181110%WP23%HOMO SAPIENS                                                       2.371411\n",
      "B CELL PROLIFERATION%GOBP%GO:0042100                                                                                            2.341091\n",
      "B CELL ACTIVATION%PANTHER PATHWAY%P00010                                                                                        2.338723\n",
      "CD22 MEDIATED BCR REGULATION%REACTOME%R-HSA-5690714.2                                                                           2.334895\n",
      "PHAGOCYTOSIS, RECOGNITION%GOBP%GO:0006910                                                                                       2.329463\n",
      "IMMUNE RESPONSE-ACTIVATING CELL SURFACE RECEPTOR SIGNALING PATHWAY%GOBP%GO:0002429                                              2.328828\n",
      "T-CELL ANTIGEN RECEPTOR (TCR) PATHWAY DURING STAPHYLOCOCCUS AUREUS INFECTION%WIKIPATHWAYS_20181110%WP3863%HOMO SAPIENS          2.327103\n",
      "IMMUNE RESPONSE-REGULATING CELL SURFACE RECEPTOR SIGNALING PATHWAY%GOBP%GO:0002768                                              2.320066\n",
      "LYMPHOCYTE DIFFERENTIATION%GOBP%GO:0030098                                                                                      2.314311\n",
      "HUMORAL IMMUNE RESPONSE MEDIATED BY CIRCULATING IMMUNOGLOBULIN%GOBP%GO:0002455                                                  2.309377\n",
      "INTERLEUKIN-3, INTERLEUKIN-5 AND GM-CSF SIGNALING%REACTOME%R-HSA-512988.5                                                       2.306678\n",
      "BIOCARTA_TCRA_PATHWAY%MSIGDB_C2%BIOCARTA_TCRA_PATHWAY                                                                           2.299730\n",
      "PID_PI3KCI_PATHWAY%MSIGDB_C2%PID_PI3KCI_PATHWAY                                                                                 2.293670\n",
      "B CELL DIFFERENTIATION%GOBP%GO:0030183                                                                                          2.292818\n",
      "IMMUNE RESPONSE-ACTIVATING SIGNAL TRANSDUCTION%GOBP%GO:0002757                                                                  2.277341\n",
      "IMMUNE RESPONSE-REGULATING SIGNALING PATHWAY%GOBP%GO:0002764                                                                    2.265846\n",
      "FCERI MEDIATED MAPK ACTIVATION%REACTOME DATABASE ID RELEASE 66%2871796                                                          2.265550\n",
      "T CELL ACTIVATION%PANTHER PATHWAY%P00053                                                                                        2.264939\n",
      "BIOCARTA_THELPER_PATHWAY%MSIGDB_C2%BIOCARTA_THELPER_PATHWAY                                                                     2.263595\n",
      "BIOCARTA_TCYTOTOXIC_PATHWAY%MSIGDB_C2%BIOCARTA_TCYTOTOXIC_PATHWAY                                                               2.259509\n",
      "LEUKOCYTE ADHESION TO VASCULAR ENDOTHELIAL CELL%GOBP%GO:0061756                                                                 2.253362\n",
      "LEUKOCYTE TETHERING OR ROLLING%GOBP%GO:0050901                                                                                  2.253362\n",
      "T-CELL ANTIGEN RECEPTOR (TCR)  SIGNALING PATHWAY%WIKIPATHWAYS_20181110%WP69%HOMO SAPIENS                                        2.250678\n",
      "STIMULATORY C-TYPE LECTIN RECEPTOR SIGNALING PATHWAY%GOBP%GO:0002223                                                            2.237091\n",
      "ADAPTIVE IMMUNE RESPONSE%GOBP%GO:0002250                                                                                        2.210384\n",
      "INNATE IMMUNE RESPONSE ACTIVATING CELL SURFACE RECEPTOR SIGNALING PATHWAY%GOBP%GO:0002220                                       2.197837\n",
      "CELLULAR EXTRAVASATION%GOBP%GO:0045123                                                                                          2.193573\n",
      "SIG_BCR_SIGNALING_PATHWAY%MSIGDB_C2%SIG_BCR_SIGNALING_PATHWAY                                                                   2.191334\n",
      "REGULATION OF B CELL RECEPTOR SIGNALING PATHWAY%GOBP%GO:0050855                                                                 2.179292\n",
      "BIOCARTA_CSK_PATHWAY%MSIGDB_C2%BIOCARTA_CSK_PATHWAY                                                                             2.175294\n",
      "CLASS I PI3K SIGNALING EVENTS%PATHWAY INTERACTION DATABASE NCI-NATURE CURATED DATA%CLASS I PI3K SIGNALING EVENTS                2.174673\n",
      "LEUKOCYTE CELL-CELL ADHESION%GOBP%GO:0007159                                                                                    2.174003\n",
      "BIOCARTA_CTLA4_PATHWAY%MSIGDB_C2%BIOCARTA_CTLA4_PATHWAY                                                                         2.171473\n",
      "Name: NES, dtype: float64\n",
      "\n",
      "\n",
      "pathway\n",
      "HALLMARK_UV_RESPONSE_DN%MSIGDB_C2%HALLMARK_UV_RESPONSE_DN                                                                                             2.622203\n",
      "ERBB1 DOWNSTREAM SIGNALING%PATHWAY INTERACTION DATABASE NCI-NATURE CURATED DATA%ERBB1 DOWNSTREAM SIGNALING                                            2.479258\n",
      "BIOCARTA_ALK_PATHWAY%MSIGDB_C2%BIOCARTA_ALK_PATHWAY                                                                                                   2.387594\n",
      "REGULATION OF CELL JUNCTION ASSEMBLY%GOBP%GO:1901888                                                                                                  2.381998\n",
      "REGULATION OF EPITHELIAL CELL MIGRATION%GOBP%GO:0010632                                                                                               2.376502\n",
      "PID_ERBB1_DOWNSTREAM_PATHWAY%MSIGDB_C2%PID_ERBB1_DOWNSTREAM_PATHWAY                                                                                   2.347470\n",
      "HALLMARK_PROTEIN_SECRETION%MSIGDB_C2%HALLMARK_PROTEIN_SECRETION                                                                                       2.343975\n",
      "REGULATION OF CELL-SUBSTRATE ADHESION%GOBP%GO:0010810                                                                                                 2.333373\n",
      "BIOCARTA_NO1_PATHWAY%MSIGDB_C2%BIOCARTA_NO1_PATHWAY                                                                                                   2.331084\n",
      "ACYL-COA BIOSYNTHETIC PROCESS%GOBP%GO:0071616                                                                                                         2.316843\n",
      "THIOESTER BIOSYNTHETIC PROCESS%GOBP%GO:0035384                                                                                                        2.316843\n",
      "REGULATION OF STEM CELL DIFFERENTIATION%GOBP%GO:2000736                                                                                               2.313732\n",
      "SIGNALING BY HIPPO%REACTOME DATABASE ID RELEASE 66%2028269                                                                                            2.297513\n",
      "MICRORNAS IN CARDIOMYOCYTE HYPERTROPHY%WIKIPATHWAYS_20181110%WP1544%HOMO SAPIENS                                                                      2.293160\n",
      "NUCLEOSIDE BISPHOSPHATE BIOSYNTHETIC PROCESS%GOBP%GO:0033866                                                                                          2.281124\n",
      "PURINE NUCLEOSIDE BISPHOSPHATE BIOSYNTHETIC PROCESS%GOBP%GO:0034033                                                                                   2.281124\n",
      "RIBONUCLEOSIDE BISPHOSPHATE BIOSYNTHETIC PROCESS%GOBP%GO:0034030                                                                                      2.281124\n",
      "VASOPRESSIN SYNTHESIS%PANTHER PATHWAY%P04395                                                                                                          2.266858\n",
      "FATTY-ACYL-COA METABOLIC PROCESS%GOBP%GO:0035337                                                                                                      2.266263\n",
      "POST-GOLGI VESICLE-MEDIATED TRANSPORT%GOBP%GO:0006892                                                                                                 2.262691\n",
      "SIGNALING BY FGFR1 IN DISEASE%REACTOME DATABASE ID RELEASE 66%5655302                                                                                 2.256782\n",
      "SIGNALING EVENTS MEDIATED BY VEGFR1 AND VEGFR2%PATHWAY INTERACTION DATABASE NCI-NATURE CURATED DATA%SIGNALING EVENTS MEDIATED BY VEGFR1 AND VEGFR2    2.245784\n",
      "FOXO FAMILY SIGNALING%PATHWAY INTERACTION DATABASE NCI-NATURE CURATED DATA%FOXO FAMILY SIGNALING                                                      2.241820\n",
      "VEGFA-VEGFR2 PATHWAY%REACTOME DATABASE ID RELEASE 66%4420097                                                                                          2.239558\n",
      "LYSOSOMAL TRANSPORT%GOBP%GO:0007041                                                                                                                   2.236245\n",
      "VESICLE CYTOSKELETAL TRAFFICKING%GOBP%GO:0099518                                                                                                      2.233214\n",
      "FATTY-ACYL-COA BIOSYNTHETIC PROCESS%GOBP%GO:0046949                                                                                                   2.231833\n",
      "REGULATION OF ADHERENS JUNCTION ORGANIZATION%GOBP%GO:1903391                                                                                          2.230394\n",
      "REGULATION OF RAS FAMILY ACTIVATION%PATHWAY INTERACTION DATABASE NCI-NATURE CURATED DATA%REGULATION OF RAS FAMILY ACTIVATION                          2.230252\n",
      "HALLMARK_ANGIOGENESIS%MSIGDB_C2%HALLMARK_ANGIOGENESIS                                                                                                 2.216731\n",
      "PI-3K CASCADE:FGFR3%REACTOME DATABASE ID RELEASE 66%5654710                                                                                           2.214332\n",
      "SIGNALING BY VEGF%REACTOME DATABASE ID RELEASE 66%194138                                                                                              2.214171\n",
      "DOWNSTREAM SIGNALING OF ACTIVATED FGFR1%REACTOME DATABASE ID RELEASE 66%5654687                                                                       2.213380\n",
      "REGULATION OF CELL-SUBSTRATE JUNCTION ASSEMBLY%GOBP%GO:0090109                                                                                        2.209404\n",
      "REGULATION OF FOCAL ADHESION ASSEMBLY%GOBP%GO:0051893                                                                                                 2.209404\n",
      "FOCAL ADHESION%WIKIPATHWAYS_20181110%WP306%HOMO SAPIENS                                                                                               2.201935\n",
      "DOWNSTREAM SIGNALING OF ACTIVATED FGFR3%REACTOME%R-HSA-5654708.1                                                                                      2.199432\n",
      "PID_ECADHERIN_STABILIZATION_PATHWAY%MSIGDB_C2%PID_ECADHERIN_STABILIZATION_PATHWAY                                                                     2.195776\n",
      "HIPPO SIGNALING%GOBP%GO:0035329                                                                                                                       2.195123\n",
      "PHYSIOLOGICAL AND PATHOLOGICAL HYPERTROPHY  OF THE HEART%WIKIPATHWAYS_20181110%WP1528%HOMO SAPIENS                                                    2.193491\n",
      "PLATELET HOMEOSTASIS%REACTOME DATABASE ID RELEASE 66%418346                                                                                           2.192943\n",
      "PID_FAK_PATHWAY%MSIGDB_C2%PID_FAK_PATHWAY                                                                                                             2.192768\n",
      "SUBSTRATE ADHESION-DEPENDENT CELL SPREADING%GOBP%GO:0034446                                                                                           2.192691\n",
      "PLASMA MEMBRANE ORGANIZATION%GOBP%GO:0007009                                                                                                          2.190487\n",
      "TGF-BETA RECEPTOR SIGNALING%WIKIPATHWAYS_20181110%WP560%HOMO SAPIENS                                                                                  2.189973\n",
      "ANGIOPOIETIN RECEPTOR TIE2-MEDIATED SIGNALING%PATHWAY INTERACTION DATABASE NCI-NATURE CURATED DATA%ANGIOPOIETIN RECEPTOR TIE2-MEDIATED SIGNALING      2.188028\n",
      "POSITIVE REGULATION OF EPITHELIAL CELL MIGRATION%GOBP%GO:0010634                                                                                      2.187854\n",
      "HALLMARK_ANDROGEN_RESPONSE%MSIGDB_C2%HALLMARK_ANDROGEN_RESPONSE                                                                                       2.186670\n",
      "PID_FOXO_PATHWAY%MSIGDB_C2%PID_FOXO_PATHWAY                                                                                                           2.185755\n",
      "POSITIVE REGULATION OF CELL JUNCTION ASSEMBLY%GOBP%GO:1901890                                                                                         2.184127\n",
      "Name: NES, dtype: float64\n",
      "\n",
      "\n",
      "\n"
     ]
    },
    {
     "ename": "NameError",
     "evalue": "name 'cluster1_metastatic' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m\u001b[0m",
      "\u001b[0;31mNameError\u001b[0mTraceback (most recent call last)",
      "\u001b[0;32m<ipython-input-7-e721d3ddb6fd>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m     34\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     35\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mf2\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mintersection\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0minterest\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 36\u001b[0;31m                 \u001b[0mcluster1_metastatic\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msamp\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     37\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     38\u001b[0m         \u001b[0msub_cohort\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mcluster\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mextend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlist\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mf1\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvalues\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'cluster1_metastatic' is not defined"
     ]
    }
   ],
   "source": [
    "_dir = '../data/gsea/'\n",
    "\n",
    "sub_cohort = collections.defaultdict(list)\n",
    "sub_cohort_unique = collections.defaultdict(list)\n",
    "sub_clust_unique = collections.defaultdict(list)\n",
    "\n",
    "sigs = set()\n",
    "\n",
    "interest = [\"REGULATION OF VASCULATURE DEVELOPMENT%GOBP%GO:1901342\", \n",
    "            \"POSITIVE REGULATION OF LOCOMOTION%GOBP%GO:0040017\",\n",
    "            \"PLATELET DEGRANULATION%GOBP%GO:0002576\",\n",
    "            \"RESPONSE TO ELEVATED PLATELET CYTOSOLIC CA2+%REACTOME DATABASE ID RELEASE 66%76005\",\n",
    "            \"POSITIVE REGULATION OF VASCULATURE DEVELOPMENT%GOBP%GO:1904018\"]\n",
    "\n",
    "for cluster, rows in assign.groupby('1'):\n",
    "    samps = rows.index.values\n",
    "    for samp in samps:\n",
    "        pth = os.path.join(_dir, 'sub_cohort', str(cluster), samp)        \n",
    "        f1 = pd.read_csv(pth, index_col=0)\n",
    "        f1 = f1[(f1['padj'] < 0.05) & (f1['ES'] > 0.0)]\n",
    "        sigs.update(f1.index.values)\n",
    "        \n",
    "        pth = os.path.join(_dir, 'sub_clust', str(cluster), samp)\n",
    "        f2 = pd.read_csv(pth, index_col=0)\n",
    "        f2 = f2[(f2['padj'] < 0.05) & (f2['ES'] > 0.0)]\n",
    "        sigs.update(f2.index.values)\n",
    "        \n",
    "        if cluster == 1:\n",
    "            print samp\n",
    "            print f1.sort_values('NES', ascending=False)['NES'].head(50)\n",
    "            print '\\n'\n",
    "            print f2.sort_values('NES', ascending=False)['NES'].head(50)\n",
    "            print '\\n\\n'\n",
    "            \n",
    "            if len(f2.index.intersection(interest)) > 0:\n",
    "                cluster1_metastatic.append(samp)\n",
    "        \n",
    "        sub_cohort[cluster].extend(list(f1.index.values))\n",
    "        sub_cohort_unique[cluster].extend(list(set(f1.index.values) - set(f2.index.values)))\n",
    "        sub_clust_unique[cluster].extend(list(set(f2.index.values) - set(f1.index.values)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2323"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(sigs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 31\n",
      "T CELL SELECTION%GOBP%GO:0045058 4\n",
      "PLASMA MEMBRANE INVAGINATION%GOBP%GO:0099024 4\n",
      "THYMIC T CELL SELECTION%GOBP%GO:0045061 4\n",
      "BIOCARTA_TCRA_PATHWAY%MSIGDB_C2%BIOCARTA_TCRA_PATHWAY 4\n",
      "B CELL MEDIATED IMMUNITY%GOBP%GO:0019724 4\n",
      "POSITIVE REGULATION OF B CELL ACTIVATION%GOBP%GO:0050871 4\n",
      "PRODUCTION OF MOLECULAR MEDIATOR OF IMMUNE RESPONSE%GOBP%GO:0002440 4\n",
      "COMPLEMENT ACTIVATION, CLASSICAL PATHWAY%GOBP%GO:0006958 4\n",
      "PHAGOCYTOSIS, ENGULFMENT%GOBP%GO:0006911 4\n",
      "BIOCARTA_TCYTOTOXIC_PATHWAY%MSIGDB_C2%BIOCARTA_TCYTOTOXIC_PATHWAY 4\n",
      "TCR SIGNALING IN NAIVE CD8+ T CELLS%PATHWAY INTERACTION DATABASE NCI-NATURE CURATED DATA%TCR SIGNALING IN NAIVE CD8+ T CELLS 4\n",
      "COMPLEMENT ACTIVATION%GOBP%GO:0006956 4\n",
      "GENERATION OF SECOND MESSENGER MOLECULES%REACTOME DATABASE ID RELEASE 66%202433 4\n",
      "IMMUNOGLOBULIN PRODUCTION%GOBP%GO:0002377 4\n",
      "HUMORAL IMMUNE RESPONSE MEDIATED BY CIRCULATING IMMUNOGLOBULIN%GOBP%GO:0002455 4\n",
      "REGULATION OF DENDRITIC CELL DENDRITE ASSEMBLY%GOBP%GO:2000547 4\n",
      "B CELL PROLIFERATION%GOBP%GO:0042100 4\n",
      "BIOCARTA_THELPER_PATHWAY%MSIGDB_C2%BIOCARTA_THELPER_PATHWAY 4\n",
      "B CELL RECEPTOR SIGNALING PATHWAY%GOBP%GO:0050853 4\n",
      "PID_CD8_TCR_PATHWAY%MSIGDB_C2%PID_CD8_TCR_PATHWAY 4\n",
      "POSITIVE THYMIC T CELL SELECTION%GOBP%GO:0045059 4\n",
      "T CELL DIFFERENTIATION%GOBP%GO:0030217 4\n",
      "ESTABLISHMENT OF T CELL POLARITY%GOBP%GO:0001768 4\n",
      "IMMUNOGLOBULIN MEDIATED IMMUNE RESPONSE%GOBP%GO:0016064 4\n",
      "PHAGOCYTOSIS, RECOGNITION%GOBP%GO:0006910 4\n",
      "MEMBRANE INVAGINATION%GOBP%GO:0010324 4\n",
      "IMMUNOLOGICAL SYNAPSE FORMATION%GOBP%GO:0001771 4\n",
      "POSITIVE T CELL SELECTION%GOBP%GO:0043368 4\n",
      "MYELOID DENDRITIC CELL CHEMOTAXIS%GOBP%GO:0002408 3\n",
      "POSITIVE REGULATION OF LEUKOCYTE PROLIFERATION%GOBP%GO:0070665 3\n",
      "LYMPHOCYTE ACTIVATION INVOLVED IN IMMUNE RESPONSE%GOBP%GO:0002285 3\n",
      "T CELL COSTIMULATION%GOBP%GO:0031295 3\n",
      "PD-1 SIGNALING%REACTOME%R-HSA-389948.1 3\n",
      "INTERLEUKIN-12 PRODUCTION%GOBP%GO:0032615 3\n",
      "BIOCARTA_NO2IL12_PATHWAY%MSIGDB_C2%BIOCARTA_NO2IL12_PATHWAY 3\n",
      "PROTEIN ACTIVATION CASCADE%GOBP%GO:0072376 3\n",
      "POSITIVE REGULATION OF T CELL ACTIVATION%GOBP%GO:0050870 3\n",
      "LYMPHOCYTE DIFFERENTIATION%GOBP%GO:0030098 3\n",
      "LYMPHOCYTE COSTIMULATION%GOBP%GO:0031294 3\n",
      "LEUKOCYTE CELL-CELL ADHESION%GOBP%GO:0007159 3\n",
      "U12 DEPENDENT SPLICING%REACTOME%R-HSA-72165.1 3\n",
      "POSITIVE REGULATION OF LEUKOCYTE CELL-CELL ADHESION%GOBP%GO:1903039 3\n",
      "T CELL ACTIVATION%GOBP%GO:0042110 3\n",
      "TRANSLOCATION OF ZAP-70 TO IMMUNOLOGICAL SYNAPSE%REACTOME DATABASE ID RELEASE 66%202430 3\n",
      "INTERLEUKIN-12 SECRETION%GOBP%GO:0072610 3\n",
      "LEUKOCYTE ADHESION TO VASCULAR ENDOTHELIAL CELL%GOBP%GO:0061756 3\n",
      "B CELL ACTIVATION%GOBP%GO:0042113 3\n",
      "ER-PHAGOSOME PATHWAY%REACTOME DATABASE ID RELEASE 66%1236974 3\n",
      "POSITIVE REGULATION OF LYMPHOCYTE PROLIFERATION%GOBP%GO:0050671 3\n",
      "ANTIGEN ACTIVATES B CELL RECEPTOR (BCR) LEADING TO GENERATION OF SECOND MESSENGERS%REACTOME DATABASE ID RELEASE 66%983695 3\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for cluster, counts in sub_cohort_unique.items():\n",
    "    print cluster, len(assign[assign['1'] == cluster])\n",
    "    \n",
    "    counter = collections.Counter(counts)\n",
    "    \n",
    "    for gs, n in counter.most_common(50):\n",
    "        print gs, n\n",
    "    \n",
    "    print '\\n\\n'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "met = ['POSITIVE REGULATION OF LOCOMOTION%GOBP%GO:0040017',\n",
    "       'POSITIVE REGULATION OF CELL MIGRATION%GOBP%GO:0030335',\n",
    "       'POSITIVE REGULATION OF VASCULATURE DEVELOPMENT%GOBP%GO:1904018',\n",
    "       'REGULATION OF ENDOTHELIAL CELL MIGRATION%GOBP%GO:0010594',\n",
    "       'POSITIVE REGULATION OF CELLULAR COMPONENT MOVEMENT%GOBP%GO:0051272',\n",
    "       'POSITIVE REGULATION OF CELL MOTILITY%GOBP%GO:2000147',\n",
    "       'FOCAL ADHESION%GOCC%GO:0005925']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 31\n",
      "FOLDING OF ACTIN BY CCT TRIC%REACTOME DATABASE ID RELEASE 66%390450 2\n",
      "SYNAPSE ORGANIZATION%GOBP%GO:0050808 1\n",
      "TCA CYCLE%PANTHER PATHWAY%P00051 1\n",
      "BIOCARTA_EIF_PATHWAY%MSIGDB_C2%BIOCARTA_EIF_PATHWAY 1\n",
      "COOPERATION OF PDCL (PHLP1) AND TRIC CCT IN G-PROTEIN BETA FOLDING%REACTOME%R-HSA-6814122.2 1\n",
      "NCRNA METABOLIC PROCESS%GOBP%GO:0034660 1\n",
      "REGULATION OF CELLULAR PROTEIN CATABOLIC PROCESS%GOBP%GO:1903362 1\n",
      "BIOCARTA_CK1_PATHWAY%MSIGDB_C2%BIOCARTA_CK1_PATHWAY 1\n",
      "SULFUR AMINO ACID METABOLISM%REACTOME DATABASE ID RELEASE 66%1614635 1\n",
      "MONOCYTE DIFFERENTIATION%GOBP%GO:0030224 1\n",
      "COPII VESICLE COATING%GOBP%GO:0048208 1\n",
      "CENTRAL NERVOUS SYSTEM DEVELOPMENT%GOBP%GO:0007417 1\n",
      "RIBONUCLEOSIDE DIPHOSPHATE METABOLIC PROCESS%GOBP%GO:0009185 1\n",
      "ESTABLISHMENT OF INTEGRATED PROVIRAL LATENCY%GOBP%GO:0075713 1\n",
      "GOLGI VESICLE BUDDING%GOBP%GO:0048194 1\n",
      "POSITIVE REGULATION OF TRANSLATION%GOBP%GO:0045727 1\n",
      "BRAIN-DERIVED NEUROTROPHIC FACTOR (BDNF) SIGNALING PATHWAY%WIKIPATHWAYS_20181110%WP2380%HOMO SAPIENS 1\n",
      "RESPIRATORY ELECTRON TRANSPORT%REACTOME DATABASE ID RELEASE 66%611105 1\n",
      "PURINE RIBONUCLEOTIDE BIOSYNTHETIC PROCESS%GOBP%GO:0009152 1\n",
      "PENTOSE PHOSPHATE PATHWAY%PANTHER PATHWAY%P02762 1\n",
      "BETA2 ADRENERGIC RECEPTOR SIGNALING PATHWAY%PANTHER PATHWAY%P04378 1\n",
      "PYRUVATE METABOLISM AND CITRIC ACID (TCA) CYCLE%REACTOME%R-HSA-71406.1 1\n",
      "PENTOSE PHOSPHATE PATHWAY (OXIDATIVE BRANCH)%HUMANCYC%OXIDATIVEPENT-PWY 1\n",
      "REGULATION OF CELLULAR AMIDE METABOLIC PROCESS%GOBP%GO:0034248 1\n",
      "OXIDATIVE PHOSPHORYLATION%WIKIPATHWAYS_20181110%WP623%HOMO SAPIENS 1\n",
      "HUNTINGTON DISEASE%PANTHER PATHWAY%P00029 1\n",
      "COENZYME A BIOSYNTHETIC PROCESS%GOBP%GO:0015937 1\n",
      "REGULATION OF RENAL SYSTEM PROCESS%GOBP%GO:0098801 1\n",
      "NICOTINAMIDE NUCLEOTIDE METABOLIC PROCESS%GOBP%GO:0046496 1\n",
      "RIBOSE PHOSPHATE BIOSYNTHETIC PROCESS%GOBP%GO:0046390 1\n",
      "COENZYME BIOSYNTHETIC PROCESS%GOBP%GO:0009108 1\n",
      "ORGANIC HYDROXY COMPOUND BIOSYNTHETIC PROCESS%GOBP%GO:1901617 1\n",
      "NADP METABOLIC PROCESS%GOBP%GO:0006739 1\n",
      "RIBOFLAVIN BIOSYNTHETIC PROCESS%GOBP%GO:0009231 1\n",
      "POSITIVE REGULATION OF VIRAL PROCESS%GOBP%GO:0048524 1\n",
      "POSITIVE REGULATION OF INTERLEUKIN-8 SECRETION%GOBP%GO:2000484 1\n",
      "CHOLESTEROL BIOSYNTHETIC PROCESS%GOBP%GO:0006695 1\n",
      "PENTOSE PHOSPHATE PATHWAY%WIKIPATHWAYS_20181110%WP134%HOMO SAPIENS 1\n",
      "CHOLESTEROL BIOSYNTHESIS III (VIA DESMOSTEROL)%HUMANCYC%PWY66-4 1\n",
      "BIOCARTA_GPCR_PATHWAY%MSIGDB_C2%BIOCARTA_GPCR_PATHWAY 1\n",
      "AMINO ACID METABOLISM%WIKIPATHWAYS_20181110%WP3925%HOMO SAPIENS 1\n",
      "DEREGULATED CDK5 TRIGGERS MULTIPLE NEURODEGENERATIVE PATHWAYS IN ALZHEIMER'S DISEASE MODELS%REACTOME DATABASE ID RELEASE 66%8862803 1\n",
      "PENTOSE-PHOSPHATE SHUNT%GOBP%GO:0006098 1\n",
      "VESICLE LOCALIZATION%GOBP%GO:0051648 1\n",
      "DRUG METABOLIC PROCESS%GOBP%GO:0017144 1\n",
      "SECONDARY ALCOHOL BIOSYNTHETIC PROCESS%GOBP%GO:1902653 1\n",
      "POSITIVE REGULATION OF CELLULAR AMIDE METABOLIC PROCESS%GOBP%GO:0034250 1\n",
      "ACTIVATION OF PROTEIN KINASE A ACTIVITY%GOBP%GO:0034199 1\n",
      "TRANSPORT TO THE GOLGI AND SUBSEQUENT MODIFICATION%REACTOME%R-HSA-948021.2 1\n",
      "RIBOSOME BIOGENESIS%GOBP%GO:0042254 1\n",
      "AUTOINTEGRATION RESULTS IN VIRAL DNA CIRCLES%REACTOME%R-HSA-177539.2 1\n",
      "HALLMARK_PROTEIN_SECRETION%MSIGDB_C2%HALLMARK_PROTEIN_SECRETION 1\n",
      "CELLULAR RESPIRATION%GOBP%GO:0045333 1\n",
      "NICOTINE ACTIVITY ON DOPAMINERGIC NEURONS%WIKIPATHWAYS_20181110%WP1602%HOMO SAPIENS 1\n",
      "NEGATIVE REGULATION OF EOSINOPHIL EXTRAVASATION%GOBP%GO:2000420 1\n",
      "PURINE NUCLEOSIDE DIPHOSPHATE METABOLIC PROCESS%GOBP%GO:0009135 1\n",
      "TRICARBOXYLIC ACID CYCLE%GOBP%GO:0006099 1\n",
      "RIBOSOMAL SMALL SUBUNIT ASSEMBLY%GOBP%GO:0000028 1\n",
      "PARKIN-UBIQUITIN PROTEASOMAL SYSTEM PATHWAY%WIKIPATHWAYS_20181110%WP2359%HOMO SAPIENS 1\n",
      "ER TO GOLGI ANTEROGRADE TRANSPORT%REACTOME%R-HSA-199977.3 1\n",
      "REGULATION OF SYNAPSE STRUCTURE OR ACTIVITY%GOBP%GO:0050803 1\n",
      "REGULATION OF AMINE TRANSPORT%GOBP%GO:0051952 1\n",
      "CHOLESTEROL BIOSYNTHESIS%WIKIPATHWAYS_20181110%WP197%HOMO SAPIENS 1\n",
      "PURINE NUCLEOTIDE METABOLIC PROCESS%GOBP%GO:0006163 1\n",
      "GLUCONEOGENESIS%REACTOME DATABASE ID RELEASE 66%70263 1\n",
      "REGULATION OF CREATINE TRANSMEMBRANE TRANSPORTER ACTIVITY%GOBP%GO:1905407 1\n",
      "RIBOSOMAL LARGE SUBUNIT ASSEMBLY%GOBP%GO:0000027 1\n",
      "CHOLESTEROL BIOSYNTHESIS II (VIA 24,25-DIHYDROLANOSTEROL)%HUMANCYC%PWY66-3 1\n",
      "PENTOSE PHOSPHATE PATHWAY%HUMANCYC%PENTOSE-P-PWY 1\n",
      "PID_PDGFRB_PATHWAY%MSIGDB_C2%PID_PDGFRB_PATHWAY 1\n",
      "EPHRIN RECEPTOR SIGNALING PATHWAY%GOBP%GO:0048013 1\n",
      "ROBO RECEPTORS BIND AKAP5%REACTOME%R-HSA-9010642.1 1\n",
      "FORMATION OF ATP BY CHEMIOSMOTIC COUPLING%REACTOME%R-HSA-163210.2 1\n",
      "EPH-EPHRIN SIGNALING%REACTOME%R-HSA-2682334.1 1\n",
      "REGULATION OF RESPONSE TO DRUG%GOBP%GO:2001023 1\n",
      "REGULATION OF MITOCHONDRIAL MEMBRANE POTENTIAL%GOBP%GO:0051881 1\n",
      "POSITIVE REGULATION OF PROTEIN LOCALIZATION TO CAJAL BODY%GOBP%GO:1904871 1\n",
      "CITRATE METABOLIC PROCESS%GOBP%GO:0006101 1\n",
      "OXIDATION-REDUCTION PROCESS%GOBP%GO:0055114 1\n",
      "2-LTR CIRCLE FORMATION%REACTOME%R-HSA-164843.2 1\n",
      "COPI-INDEPENDENT GOLGI-TO-ER RETROGRADE TRAFFIC%REACTOME DATABASE ID RELEASE 66%6811436 1\n",
      "STEROL BIOSYNTHETIC PROCESS%GOBP%GO:0016126 1\n",
      "MITOCHONDRIAL ATP SYNTHESIS COUPLED PROTON TRANSPORT%GOBP%GO:0042776 1\n",
      "SUPERPATHWAY OF CHOLESTEROL BIOSYNTHESIS%HUMANCYC%PWY66-5 1\n",
      "CHOLESTEROL BIOSYNTHESIS I%HUMANCYC%PWY66-341 1\n",
      "RIBONUCLEOTIDE BIOSYNTHETIC PROCESS%GOBP%GO:0009260 1\n",
      "CALCIUM REGULATION IN THE CARDIAC CELL%WIKIPATHWAYS_20181110%WP536%HOMO SAPIENS 1\n",
      "PROTEIN METHYLATION%REACTOME%R-HSA-8876725.2 1\n",
      "PROTEIN LOCALIZATION TO MITOCHONDRION%GOBP%GO:0070585 1\n",
      "BIOCARTA_ACTINY_PATHWAY%MSIGDB_C2%BIOCARTA_ACTINY_PATHWAY 1\n",
      "COPI-DEPENDENT GOLGI-TO-ER RETROGRADE TRAFFIC%REACTOME%R-HSA-6811434.1 1\n",
      "ESTABLISHMENT OF VESICLE LOCALIZATION%GOBP%GO:0051650 1\n",
      "NEURON PROJECTION DEVELOPMENT%GOBP%GO:0031175 1\n",
      "CHOLESTEROL BIOSYNTHESIS%PANTHER PATHWAY%P00014 1\n",
      "ENERGY DERIVATION BY OXIDATION OF ORGANIC COMPOUNDS%GOBP%GO:0015980 1\n",
      "SYNTHESIS, SECRETION, AND INACTIVATION OF GLUCOSE-DEPENDENT INSULINOTROPIC POLYPEPTIDE (GIP)%REACTOME DATABASE ID RELEASE 66%400511 1\n",
      "REGULATION OF MEMBRANE PERMEABILITY%GOBP%GO:0090559 1\n",
      "ELECTRON TRANSPORT CHAIN (OXPHOS SYSTEM IN MITOCHONDRIA)%WIKIPATHWAYS_20181110%WP111%HOMO SAPIENS 1\n",
      "ACTIVATION OF BH3-ONLY PROTEINS%REACTOME DATABASE ID RELEASE 66%114452 1\n",
      "HALLMARK_PANCREAS_BETA_CELLS%MSIGDB_C2%HALLMARK_PANCREAS_BETA_CELLS 1\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "top5 = []\n",
    "\n",
    "gs_plot = pd.DataFrame(columns=['cluster', 'gs', 'percent'])\n",
    "for cluster, counts in sub_clust_unique.items():\n",
    "    N = len(assign[assign['1'] == cluster])\n",
    "    \n",
    "    print cluster, N\n",
    "    \n",
    "    counter = collections.Counter(counts)\n",
    "    \n",
    "    for gs, n in counter.most_common(100):\n",
    "        print gs, n\n",
    "        if cluster == 1:\n",
    "            top5.append(gs)\n",
    "        \n",
    "    print '\\n\\n'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['FOLDING OF ACTIN BY CCT TRIC%REACTOME DATABASE ID RELEASE 66%390450',\n",
       " 'SYNAPSE ORGANIZATION%GOBP%GO:0050808',\n",
       " 'TCA CYCLE%PANTHER PATHWAY%P00051',\n",
       " 'BIOCARTA_EIF_PATHWAY%MSIGDB_C2%BIOCARTA_EIF_PATHWAY',\n",
       " 'COOPERATION OF PDCL (PHLP1) AND TRIC CCT IN G-PROTEIN BETA FOLDING%REACTOME%R-HSA-6814122.2',\n",
       " 'NCRNA METABOLIC PROCESS%GOBP%GO:0034660',\n",
       " 'REGULATION OF CELLULAR PROTEIN CATABOLIC PROCESS%GOBP%GO:1903362',\n",
       " 'BIOCARTA_CK1_PATHWAY%MSIGDB_C2%BIOCARTA_CK1_PATHWAY',\n",
       " 'SULFUR AMINO ACID METABOLISM%REACTOME DATABASE ID RELEASE 66%1614635',\n",
       " 'MONOCYTE DIFFERENTIATION%GOBP%GO:0030224',\n",
       " 'COPII VESICLE COATING%GOBP%GO:0048208',\n",
       " 'CENTRAL NERVOUS SYSTEM DEVELOPMENT%GOBP%GO:0007417',\n",
       " 'RIBONUCLEOSIDE DIPHOSPHATE METABOLIC PROCESS%GOBP%GO:0009185',\n",
       " 'ESTABLISHMENT OF INTEGRATED PROVIRAL LATENCY%GOBP%GO:0075713',\n",
       " 'GOLGI VESICLE BUDDING%GOBP%GO:0048194',\n",
       " 'POSITIVE REGULATION OF TRANSLATION%GOBP%GO:0045727',\n",
       " 'BRAIN-DERIVED NEUROTROPHIC FACTOR (BDNF) SIGNALING PATHWAY%WIKIPATHWAYS_20181110%WP2380%HOMO SAPIENS',\n",
       " 'RESPIRATORY ELECTRON TRANSPORT%REACTOME DATABASE ID RELEASE 66%611105',\n",
       " 'PURINE RIBONUCLEOTIDE BIOSYNTHETIC PROCESS%GOBP%GO:0009152',\n",
       " 'PENTOSE PHOSPHATE PATHWAY%PANTHER PATHWAY%P02762',\n",
       " 'BETA2 ADRENERGIC RECEPTOR SIGNALING PATHWAY%PANTHER PATHWAY%P04378',\n",
       " 'PYRUVATE METABOLISM AND CITRIC ACID (TCA) CYCLE%REACTOME%R-HSA-71406.1',\n",
       " 'PENTOSE PHOSPHATE PATHWAY (OXIDATIVE BRANCH)%HUMANCYC%OXIDATIVEPENT-PWY',\n",
       " 'REGULATION OF CELLULAR AMIDE METABOLIC PROCESS%GOBP%GO:0034248',\n",
       " 'OXIDATIVE PHOSPHORYLATION%WIKIPATHWAYS_20181110%WP623%HOMO SAPIENS',\n",
       " 'HUNTINGTON DISEASE%PANTHER PATHWAY%P00029',\n",
       " 'COENZYME A BIOSYNTHETIC PROCESS%GOBP%GO:0015937',\n",
       " 'REGULATION OF RENAL SYSTEM PROCESS%GOBP%GO:0098801',\n",
       " 'NICOTINAMIDE NUCLEOTIDE METABOLIC PROCESS%GOBP%GO:0046496',\n",
       " 'RIBOSE PHOSPHATE BIOSYNTHETIC PROCESS%GOBP%GO:0046390',\n",
       " 'COENZYME BIOSYNTHETIC PROCESS%GOBP%GO:0009108',\n",
       " 'ORGANIC HYDROXY COMPOUND BIOSYNTHETIC PROCESS%GOBP%GO:1901617',\n",
       " 'NADP METABOLIC PROCESS%GOBP%GO:0006739',\n",
       " 'RIBOFLAVIN BIOSYNTHETIC PROCESS%GOBP%GO:0009231',\n",
       " 'POSITIVE REGULATION OF VIRAL PROCESS%GOBP%GO:0048524',\n",
       " 'POSITIVE REGULATION OF INTERLEUKIN-8 SECRETION%GOBP%GO:2000484',\n",
       " 'CHOLESTEROL BIOSYNTHETIC PROCESS%GOBP%GO:0006695',\n",
       " 'PENTOSE PHOSPHATE PATHWAY%WIKIPATHWAYS_20181110%WP134%HOMO SAPIENS',\n",
       " 'CHOLESTEROL BIOSYNTHESIS III (VIA DESMOSTEROL)%HUMANCYC%PWY66-4',\n",
       " 'BIOCARTA_GPCR_PATHWAY%MSIGDB_C2%BIOCARTA_GPCR_PATHWAY',\n",
       " 'AMINO ACID METABOLISM%WIKIPATHWAYS_20181110%WP3925%HOMO SAPIENS',\n",
       " \"DEREGULATED CDK5 TRIGGERS MULTIPLE NEURODEGENERATIVE PATHWAYS IN ALZHEIMER'S DISEASE MODELS%REACTOME DATABASE ID RELEASE 66%8862803\",\n",
       " 'PENTOSE-PHOSPHATE SHUNT%GOBP%GO:0006098',\n",
       " 'VESICLE LOCALIZATION%GOBP%GO:0051648',\n",
       " 'DRUG METABOLIC PROCESS%GOBP%GO:0017144',\n",
       " 'SECONDARY ALCOHOL BIOSYNTHETIC PROCESS%GOBP%GO:1902653',\n",
       " 'POSITIVE REGULATION OF CELLULAR AMIDE METABOLIC PROCESS%GOBP%GO:0034250',\n",
       " 'ACTIVATION OF PROTEIN KINASE A ACTIVITY%GOBP%GO:0034199',\n",
       " 'TRANSPORT TO THE GOLGI AND SUBSEQUENT MODIFICATION%REACTOME%R-HSA-948021.2',\n",
       " 'RIBOSOME BIOGENESIS%GOBP%GO:0042254',\n",
       " 'AUTOINTEGRATION RESULTS IN VIRAL DNA CIRCLES%REACTOME%R-HSA-177539.2',\n",
       " 'HALLMARK_PROTEIN_SECRETION%MSIGDB_C2%HALLMARK_PROTEIN_SECRETION',\n",
       " 'CELLULAR RESPIRATION%GOBP%GO:0045333',\n",
       " 'NICOTINE ACTIVITY ON DOPAMINERGIC NEURONS%WIKIPATHWAYS_20181110%WP1602%HOMO SAPIENS',\n",
       " 'NEGATIVE REGULATION OF EOSINOPHIL EXTRAVASATION%GOBP%GO:2000420',\n",
       " 'PURINE NUCLEOSIDE DIPHOSPHATE METABOLIC PROCESS%GOBP%GO:0009135',\n",
       " 'TRICARBOXYLIC ACID CYCLE%GOBP%GO:0006099',\n",
       " 'RIBOSOMAL SMALL SUBUNIT ASSEMBLY%GOBP%GO:0000028',\n",
       " 'PARKIN-UBIQUITIN PROTEASOMAL SYSTEM PATHWAY%WIKIPATHWAYS_20181110%WP2359%HOMO SAPIENS',\n",
       " 'ER TO GOLGI ANTEROGRADE TRANSPORT%REACTOME%R-HSA-199977.3',\n",
       " 'REGULATION OF SYNAPSE STRUCTURE OR ACTIVITY%GOBP%GO:0050803',\n",
       " 'REGULATION OF AMINE TRANSPORT%GOBP%GO:0051952',\n",
       " 'CHOLESTEROL BIOSYNTHESIS%WIKIPATHWAYS_20181110%WP197%HOMO SAPIENS',\n",
       " 'PURINE NUCLEOTIDE METABOLIC PROCESS%GOBP%GO:0006163',\n",
       " 'GLUCONEOGENESIS%REACTOME DATABASE ID RELEASE 66%70263',\n",
       " 'REGULATION OF CREATINE TRANSMEMBRANE TRANSPORTER ACTIVITY%GOBP%GO:1905407',\n",
       " 'RIBOSOMAL LARGE SUBUNIT ASSEMBLY%GOBP%GO:0000027',\n",
       " 'CHOLESTEROL BIOSYNTHESIS II (VIA 24,25-DIHYDROLANOSTEROL)%HUMANCYC%PWY66-3',\n",
       " 'PENTOSE PHOSPHATE PATHWAY%HUMANCYC%PENTOSE-P-PWY',\n",
       " 'PID_PDGFRB_PATHWAY%MSIGDB_C2%PID_PDGFRB_PATHWAY',\n",
       " 'EPHRIN RECEPTOR SIGNALING PATHWAY%GOBP%GO:0048013',\n",
       " 'ROBO RECEPTORS BIND AKAP5%REACTOME%R-HSA-9010642.1',\n",
       " 'FORMATION OF ATP BY CHEMIOSMOTIC COUPLING%REACTOME%R-HSA-163210.2',\n",
       " 'EPH-EPHRIN SIGNALING%REACTOME%R-HSA-2682334.1',\n",
       " 'REGULATION OF RESPONSE TO DRUG%GOBP%GO:2001023',\n",
       " 'REGULATION OF MITOCHONDRIAL MEMBRANE POTENTIAL%GOBP%GO:0051881',\n",
       " 'POSITIVE REGULATION OF PROTEIN LOCALIZATION TO CAJAL BODY%GOBP%GO:1904871',\n",
       " 'CITRATE METABOLIC PROCESS%GOBP%GO:0006101',\n",
       " 'OXIDATION-REDUCTION PROCESS%GOBP%GO:0055114',\n",
       " '2-LTR CIRCLE FORMATION%REACTOME%R-HSA-164843.2',\n",
       " 'COPI-INDEPENDENT GOLGI-TO-ER RETROGRADE TRAFFIC%REACTOME DATABASE ID RELEASE 66%6811436',\n",
       " 'STEROL BIOSYNTHETIC PROCESS%GOBP%GO:0016126',\n",
       " 'MITOCHONDRIAL ATP SYNTHESIS COUPLED PROTON TRANSPORT%GOBP%GO:0042776',\n",
       " 'SUPERPATHWAY OF CHOLESTEROL BIOSYNTHESIS%HUMANCYC%PWY66-5',\n",
       " 'CHOLESTEROL BIOSYNTHESIS I%HUMANCYC%PWY66-341',\n",
       " 'RIBONUCLEOTIDE BIOSYNTHETIC PROCESS%GOBP%GO:0009260',\n",
       " 'CALCIUM REGULATION IN THE CARDIAC CELL%WIKIPATHWAYS_20181110%WP536%HOMO SAPIENS',\n",
       " 'PROTEIN METHYLATION%REACTOME%R-HSA-8876725.2',\n",
       " 'PROTEIN LOCALIZATION TO MITOCHONDRION%GOBP%GO:0070585',\n",
       " 'BIOCARTA_ACTINY_PATHWAY%MSIGDB_C2%BIOCARTA_ACTINY_PATHWAY',\n",
       " 'COPI-DEPENDENT GOLGI-TO-ER RETROGRADE TRAFFIC%REACTOME%R-HSA-6811434.1',\n",
       " 'ESTABLISHMENT OF VESICLE LOCALIZATION%GOBP%GO:0051650',\n",
       " 'NEURON PROJECTION DEVELOPMENT%GOBP%GO:0031175',\n",
       " 'CHOLESTEROL BIOSYNTHESIS%PANTHER PATHWAY%P00014',\n",
       " 'ENERGY DERIVATION BY OXIDATION OF ORGANIC COMPOUNDS%GOBP%GO:0015980',\n",
       " 'SYNTHESIS, SECRETION, AND INACTIVATION OF GLUCOSE-DEPENDENT INSULINOTROPIC POLYPEPTIDE (GIP)%REACTOME DATABASE ID RELEASE 66%400511',\n",
       " 'REGULATION OF MEMBRANE PERMEABILITY%GOBP%GO:0090559',\n",
       " 'ELECTRON TRANSPORT CHAIN (OXPHOS SYSTEM IN MITOCHONDRIA)%WIKIPATHWAYS_20181110%WP111%HOMO SAPIENS',\n",
       " 'ACTIVATION OF BH3-ONLY PROTEINS%REACTOME DATABASE ID RELEASE 66%114452',\n",
       " 'HALLMARK_PANCREAS_BETA_CELLS%MSIGDB_C2%HALLMARK_PANCREAS_BETA_CELLS']"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "top5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'cluster1_metastatic' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m\u001b[0m",
      "\u001b[0;31mNameError\u001b[0mTraceback (most recent call last)",
      "\u001b[0;32m<ipython-input-13-5d1f2118cea3>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m     53\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     54\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 55\u001b[0;31m     \u001b[0mimmune\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;36m1\u001b[0m \u001b[0;32mif\u001b[0m \u001b[0msample\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mcluster1_metastatic\u001b[0m \u001b[0;32melse\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     56\u001b[0m     \u001b[0msurv\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0msample\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m:\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0mOS\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvital\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mEFS\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mevent\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mimmune\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'cluster1_metastatic' is not defined"
     ]
    }
   ],
   "source": [
    "def get_event(event):\n",
    "    if pd.isnull(event):\n",
    "        print(\"NULL\")\n",
    "        return np.nan\n",
    "    \n",
    "    events = ['Relapse',\n",
    "              'Death', \n",
    "              'Progression',\n",
    "              'Event',\n",
    "              'Second Malignant Neoplasm']\n",
    "    \n",
    "    if event == 'Censored':\n",
    "        return 0\n",
    "    \n",
    "    elif event in events:\n",
    "        return 1 \n",
    "    \n",
    "    else:\n",
    "        raise ValueError(event)\n",
    "        \n",
    "def get_vital(vital):\n",
    "    if pd.isnull(vital):\n",
    "        print(\"NULL\")\n",
    "        return np.nan\n",
    "    \n",
    "    \n",
    "    if vital == 'Alive':\n",
    "        return 0\n",
    "    \n",
    "    elif vital == 'Dead':\n",
    "        return 1\n",
    "    \n",
    "    else:\n",
    "        raise ValueError(vital)\n",
    "\n",
    "pth = '../data/meta/TARGET_NBL_ClinicalData_Discovery_20170525.xlsx'\n",
    "clinical = pd.read_excel(pth, index_col=0)\n",
    "\n",
    "surv = pd.DataFrame(index=assign[assign['1'] == 2].index, \n",
    "                    columns=['OS', 'vital', 'EFS', 'event', 'immune'])\n",
    "\n",
    "for sample in assign[assign['1'] == 1].index:\n",
    "    root = '-'.join(sample.split('-')[:3])\n",
    "    if root not in clinical.index:\n",
    "        print \"Missing: \", sample\n",
    "        continue\n",
    "    \n",
    "    OS = clinical.loc[root, 'Overall Survival Time in Days'].item()\n",
    "    vital = get_vital(clinical.loc[root, 'Vital Status'])\n",
    "    \n",
    "    EFS = clinical.loc[root, 'Event Free Survival Time in Days'].item()\n",
    "    event = get_event(clinical.loc[root, 'First Event'])\n",
    "    \n",
    "    \n",
    "    immune = 1 if sample in cluster1_metastatic else 0\n",
    "    surv.loc[sample, :] = [OS, vital, EFS, event, immune]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import uuid\n",
    "\n",
    "# Can't figure out how to pass this dataframe\n",
    "survpth = '../data/mycn-na-tme-signal-subtract-cluster1-survival.tsv'\n",
    "surv.to_csv(survpth, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "%%R -i survpth\n",
    "\n",
    "require(dpylr)\n",
    "require(survival)\n",
    "require(survminer)\n",
    "\n",
    "surv <- read.table(survpth,\n",
    "                  sep='\\t',\n",
    "                  header=T)\n",
    "\n",
    "surv <- dplyr::as_tibble(surv)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "%%R\n",
    "\n",
    "fit <- survfit(Surv(OS, vital) ~ immune, data=surv)\n",
    "ggsurvplot(fit, conf.int=F, pval=T, risk.table=TRUE)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "%%R\n",
    "\n",
    "fit <- survfit(Surv(EFS, event) ~ immune, data=surv)\n",
    "ggsurvplot(fit, conf.int=F, pval=T, risk.table=TRUE)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
