<p><span> <strong></strong> </span><br />
Jacob Pfeil<sup>1,2*</sup>, Lauren M. Sanders<sup>1,2,3</sup>, Ioannis Anastopoulos<sup>1,2</sup> A. Geoffrey Lyle<sup>2,3</sup> Alana S. Weinstein<sup>1,2</sup> Yuanqing Xue<sup>1,2</sup> Andrew Blair<sup>1,2</sup> Holly C. Beale<sup>2,3</sup> Alex Lee<sup>4</sup> Stanley G. Leung<sup>4</sup> Phuong T. Dinh<sup>4</sup> Avanthi Tayi Shah<sup>4</sup> Marcus R. Breese<sup>4</sup> W. Patrick Devine<sup>5</sup> Isabel Bjork<sup>2</sup> Sofie R. Salama<sup>1,2,6</sup> E. Alejandro Sweet-Cordero<sup>4</sup> David Haussler<sup>1,2,6</sup> Olena Morozova Vaske<sup>2,3</sup><br />
<strong>1</strong> Department of Biomolecular Engineering, University of California, Santa Cruz, California, United States of America<br />
<strong>2</strong> Genomics Institute, University of California, Santa Cruz, California, United States of America<br />
<strong>3</strong> Department of Molecular, Cell and Developmental Biology, University of California, Santa Cruz, California, United States of America<br />
<strong>4</strong> Department of Pediatrics, Division of Hematology and Oncology, University of California, San Francisco, California, United States of America<br />
<strong>5</strong> Department of Anatomic Pathology, University of California, San Francisco, California, United States of America<br />
<strong>6</strong> Howard Hughes Medical Institute, University of California, Santa Cruz, California, United States of America<br />
</p>
<p>* Corresponding author: jpfeil@ucsc.edu</p>
<h1 id="abstract" class="unnumbered">Abstract</h1>
<p>Precision oncology has primarily relied on coding mutation status as a readout to define potential therapeutic benefit. Incorporation of transcriptome analysis into precision oncology workflows has proven to be challenging, as relative rather than absolute gene expression level needs to be considered, requiring differential expression analysis across samples. However, cell-of-origin and tumor microenvironment effects limit the effectiveness of these approaches. To address these challenges, we developed an unsupervised clustering approach for discovering differential pathway expression within cancer cohorts using gene expression measurements. The hydra approach models differential expression as a multimodal distribution and uses a Dirichlet Process Mixture Model to automatically infer the number of clusters from the data. Multivariate clustering of multimodally expressed genes reveals tumor expression subtypes. We demonstrate that the hydra approach is more sensitive than widely-used gene set enrichment approaches for detecting multimodal expression signatures. We applied the hydra analysis framework to high-risk neuroblastoma and a cohort of small round blue cell tumors. We discovered expression signatures associated with changes in the tumor microenvironment. These expression signatures were consistent with pathology review of the H&amp;E slide images. Furthermore, we identified an association between ATRX deletions and elevated immune marker expression in high-risk neuroblastoma samples. The hydra software is available as a Docker container for easy deployment (<a href="https://hub.docker.com/r/jpfeil/hydra">https://hub.docker.com/r/jpfeil/hydra</a>). The source code is available on GitHub (<a href="www.github.com/jpfeil/hydra">www.github.com/jpfeil/hydra</a>).</p>
<h1 id="author-summary" class="unnumbered">Author summary</h1>
<p>Pediatric cancers tend to have a small number of somatic mutations. To increase the number of actionable leads, precision pediatric oncology initiatives also analyze tumor gene expression patterns. Currently available approaches for clinical gene expression data often use arbitrary thresholds for assessing over-expression and assume gene expression is normally distributed. We show that important gene expression signatures across small round blue cell tumors are not normally distributed and are better modeled as a mixture of normal distributions. We found significant differences in the expression of proliferative signaling, extra-cellular matrix organization, and immune signaling pathways. Expression signatures correlated with differences in patient outcomes in three disease for which we had patient outcome data. The low mutation rate in pediatric cancers have led some to suggest that pediatric cancers are less immunogenic, but our analysis suggests that similar immune activation signals can be identified across pediatric cancers, so further research into pediatric tumor immunology may reveal opportunities for powerful immunotherapies.</p>
<h1 id="introduction" class="unnumbered">Introduction</h1>
<p>Large cancer sequencing projects, including The Cancer Genome Atlas (TCGA) and Therapeutically Applicable Research to Generate Effective Treatments (TARGET), have facilitated the development of cancer gene expression compendia <span class="citation" data-cites="vivianToilEnablesReproducible2017 pughGeneticLandscapeHighrisk2013 goldmanUCSCXenaPlatform2018 thecancergenomeatlasresearchnetworkCancerGenomeAtlas2013 newtonTumorMapExploringMolecular2017"></span> but these compendia often lack corresponding normal tissue expression data. Without the the normal comparator, Hoadley et al. (2018) found that cell-of-origin signals drive integrative clustering of TCGA data, with the exception of tumors that have high immune and stromal expression clustering irrespective of the cell-of-origin. Strong cell-of-origin and tumor microenvironment (TME) signals may complicate the interpretation of gene expression results for precision oncology applications, so careful modeling of the data is necessary to infer accurate conclusions from the data.</p>
<p>The TME includes tumor cells, stromal fibroblasts, infiltrating immune cells, and vasculature <span class="citation" data-cites="joyceCellExclusionImmune2015"></span>. Similarities in TME composition across tumor samples have led to the identification of TME states (i.e. inflamed, immune-excluded, immune-desert). These states are dynamic and may change as the tumor progresses, but may shed light on the immunogenicity of tumor cells and correlate with response to cancer immunotherapies <span class="citation" data-cites="chenElementsCancerImmunity2017"></span>. The TME cellular composition can be inferred from gene expression data since host cell RNA is sequenced along with the cancer cell RNA. Not accounting for the immune signal when performing gene expression analysis may confound the interpretation of clustering results <span class="citation" data-cites="aranXCellDigitallyPortraying2017 rheeImpactTumorPurity2018 hoadleyCellofOriginPatternsDominate2018"></span>.</p>
<p>Tumor progression and response to therapies is associated with features of the TME, so targeting the TME therapeutically may improve the treatment of some cancers. Immunotherapies that activate the host immune system to eradicate tumors have improved the treatment of several cancer types <span class="citation" data-cites="mellmanCancerImmunotherapyComes2011 pageImmuneModulationCancer2014"></span>. Pediatric cancers are thought to be less immunogenic because pediatric cancers have lower mutation burdens in general. Gene overexpression is another source of tumor-specific antigens which may play an important role in pediatric tumor immunogenicity. In addition to infiltrating immune cells, cancer-associated fibroblasts assist in extracellular matrix remodeling and activation of growth factor signaling that facilitate tumor growth, metastasis, and resistance to therapies. Methods are needed to both infer and characterize gene expression subtypes that correlate with tumor microenvironment states to accelerate the development of novel therapies for pediatric cancers.</p>
<p>Tumor/normal differential expression analysis in which a cohort of tumor tissues is compared to normal counterparts is an effective approach for identifying gene expression biomarkers <span class="citation" data-cites="andersCountbasedDifferentialExpression2013 andersDifferentialExpressionAnalysis2010 sonesonComparisonMethodsDifferential2013"></span>, but it is often not possible to conduct this analysis in a clinical setting. Sufficient biological and technical replicates are limited by tumor tissue availability, and healthy neighboring tissue often cannot be isolated. In addition, for many pediatric cancers, the cell-of-origin and thus the appropriate reference normal tissue is not known. Besides differential expression analysis, single-sample pathway analysis can be used to identify upregulation of biological gene sets in tumor subtypes. The most widely used pathway analysis approach is gene set enrichment analysis (GSEA). GSEA identifies coordinated expression of pathway genes using gene ranks and a Kolmogorov-Smirnov-like test statistic <span class="citation" data-cites="subramanianGeneSetEnrichment2005 moothaPGC1alpharesponsiveGenesInvolved2003"></span>. GSEA is usually performed on differentially expressed genes, but single-sample versions exist for identifying cancer subtypes. GSEA uses curated pathway gene sets like those in the Molecular Signatures Database (MSigDB) <span class="citation" data-cites="liberzonMolecularSignaturesDatabase2011"></span>.</p>
<p>Cancer gene expression subtypes are traditionally identified using unsupervised clustering analysis such as consensus clustering analysis <span class="citation" data-cites="oyeladeClusteringAlgorithmsTheir2016 johnM3CMonteCarlo2018 wilkersonConsensusClusterPlusClassDiscovery2010"></span>. These methods are generally underpowered because the number of genes greatly exceeds the number of samples. Dimensionality reduction approaches such as Principal Component Analysis (PCA) have been found to underestimate the dimensionality of gene expression data. Lenz at al. (2016) found two cases in which PCA fails to identify a biological signal: when the size of the cluster is small and when the effect size is small. Lenz et al. (2016) suggest investigating multimodally expressed genes to improve identification of subtypes. Cancer subtypes naturally lead to multimodal expression patterns where each subtype expresses genes with distinct levels and correlation patterns. Furthermore, enriching for multimodally expressed genes before clustering has been shown to improve correlations with clinical features of interest <span class="citation" data-cites="yiliMultimodalityCriterionFeature2005"></span>.</p>
<p>Gaussian mixture models are a powerful class of unsupervised clustering algorithms for detecting multimodally expressed genes <span class="citation" data-cites="ghoshMixtureModelsAssessing2004 dahlModelBasedClusteringExpression2006 kimVariableSelectionClustering2006"></span>. A Gaussian mixture model is appropriate when the expression data can be modeled by two or more Gaussian distributions <span class="citation" data-cites="gelman2013bayesian"></span>. One limitation of Gaussian mixture models in this context is that the number of clusters in the data is often not known beforehand, so a parameter search is used to identify the best-performing model. However, this is a computationally expensive approach. This problem can be overcome by placing a Dirichlet process prior on the number of expression clusters. The number of clusters is then inferred while fitting the mixture model using Markov chain Monte Carlo sampling <span class="citation" data-cites="gelman2013bayesian"></span>. This approach has not been widely used in clinical cancer research because these algorithms are computationally intensive, but recent advances in approximate sampling methods make this approach scalable for precision oncology applications <span class="citation" data-cites="thallBayesianNonparametricStatistics2017"></span>.</p>
<p>Here, we describe the hydra approach for identifying clinically relevant expression subtypes and classifying N-of-1 tumor samples. We provide an overview of the hydra framework, assess performance for detecting differential pathway expression, and apply the approach to better understand expression patterns in high-risk neuroblastoma and other small round blue cell tumors. We applied the learned models trained on publically available cancer gene expression data to the N-of-1 setting and show that this framework can identify distinct immune and stromal expression signatures that differentiate pediatric cancer samples. Finally, we discover recurrent tumor microenvironment signatures across pediatric cancer types.</p>
<h1 id="materials-and-methods" class="unnumbered">Materials and methods</h1>
<h2 id="dirichlet-process-gaussian-mixture-model">Dirichlet Process Gaussian Mixture Model</h2>
<p>Traditional parametric models, like the finite mixture model, use a fixed number of parameters (i.e. number of clusters). Over- or underfitting can occur when the parametric model does not reflect the underlying data <span class="citation" data-cites="teh2010dirichlet"></span>. Unlike the finite mixture model, the Dirichlet process mixture model (DPMM) represents a theoretically infinite number of clusters and can adapt the number of clusters based on prior belief and the data <span class="citation" data-cites="gelmanBayesianDataAnalysis2013 antoniakMixturesDirichletProcesses1974"></span>.</p>
<p>The Dirichlet process is an infinite dimensional extension of the Dirichlet distribution <span class="citation" data-cites="fergusonBayesianAnalysisNonparametric1973"></span> and is commonly used as a prior distribution for infinite mixture models <span class="citation" data-cites="muller2004nonparametric gorurDirichletProcessGaussian2010"></span>. The Dirichlet process has two parameters: the concentration parameter <span class="math inline"><em>Œ±</em></span> and centering distribution <span class="math inline"><em>H</em></span>. The concentration parameter <span class="math inline"><em>Œ±</em></span> controls the extent to which samples from the DP resemble the centering distribution <span class="math inline"><em>H</em></span>. We model gene expression as a multivariate Gaussian distribution, so our centering distribution is a normal-Wishart distribution <span class="math inline">ùí©ùí≤<sub>0</sub></span>.</p>
<p>We briefly describe the stick-breaking construction of the Dirichlet process <span class="math inline"><em>G</em>‚ÄÑ‚àº‚ÄÑ<em>D</em><em>P</em>(<em>Œ±</em>,‚ÄÜ<em>H</em>)</span> <span class="citation" data-cites="hughes2013memoized teh2010dirichlet gelmanBayesianDataAnalysis2013 fergusonBayesianAnalysisNonparametric1973"></span>. Consider a stick of unit length. To generate an infinite number of mixing weights <span class="math inline"><em>œÄ</em><sub><em>k</em></sub></span> for the DP, we sample breakpoints <span class="math inline"><em>ŒΩ</em>‚ÄÑ‚àà‚ÄÑ[0,‚ÄÜ1]</span> from a <span class="math inline">Beta</span> distribution and break the remainder of the stick at <span class="math inline"><em>ŒΩ</em></span>. Since a finite number of clusters is sufficient for modeling gene expression data and in practice the mixing weighs quickly approach zero, the Dirichlet process is truncated to the number of clusters <span class="math inline"><em>K</em></span>, which was shown to accurately approximate the infinite posterior for large K <span class="citation" data-cites="hughes2013memoized"></span>.</p>
<p><br /><span class="math display">$$\begin{gathered}
\label{eq:dp}
\nu \sim \text{Beta}(1, \alpha) \\
\pi_k = \nu_k \prod_{l=1}^{k-1}(1 - \nu_l)\end{gathered}$$</span><br /></p>
<p>Once we have the mixing components, we want to sample parameters from the centering distribution <span class="math inline"><em>H</em></span> weighted by <span class="math inline"><em>p</em><em>i</em><sub><em>k</em></sub></span>. If we consider a probability space <span class="math inline"><em>Œò</em></span> where <span class="math inline"><em>Œ∏</em><sub><em>k</em></sub><sup>*</sup>‚ÄÑ‚àà‚ÄÑ<em>Œò</em></span>, then <span class="math inline"><em>H</em></span> is a measure on the partitions of <span class="math inline"><em>Œò</em></span>. For our application, we will partition the parameter space <span class="math inline"><em>Œò</em></span> into finite, measurable partitions <span class="math inline"><em>B</em><sub>1</sub>,‚ÄÜ<em>B</em><sub>2</sub>,‚ÄÜ...,‚ÄÜ<em>B</em><sub><em>k</em></sub></span>.</p>
<p><br /><span class="math display">$$\begin{gathered}
\theta_k^{*} \sim H \\
G = \sum_{k=1}^{\infty} \pi_k \delta_{\theta_k^{*}} \\
(G(B_1), G(B_2), ..., G(B_k)) \sim \text{Dir}(\alpha H(B_1), \alpha H(B_2), ..., \alpha H(B_k)) \end{gathered}$$</span><br /></p>
<p>This construction generates the marginal of the Dirichlet process, which follows a Dirichlet distribution. Samples from the marginal distribution are finite, discrete, and sum to <span class="math inline">1</span> <span class="citation" data-cites="fergusonBayesianAnalysisNonparametric1973"></span>. Next, we outline how the DPMM groups gene expression samples under cluster-specific parameters <span class="math inline"><em>Œº</em><sub><em>z</em><sub><em>i</em></sub></sub></span> and <span class="math inline"><em>Œ£</em><sub><em>z</em><sub><em>i</em></sub></sub></span> where <span class="math inline"><em>z</em><sub><em>i</em></sub>‚ÄÑ‚àà‚ÄÑ1,‚ÄÜ2,‚ÄÜ...,‚ÄÜ<em>K</em></span> is the cluster index.</p>
<p><br /><span class="math display">$$\begin{gathered}
\label{eq:mm}
x_i | \mu_{z_i}, \Sigma_{z_i} \sim \mathcal{N}(\mu_{z_i}, \Sigma_{z_i}) \\
z_i | \pi \sim \text{Categorical}(\pi_1, \pi_2, ..., \pi_k) \\
\mu_{z_i}, \Sigma_{z_i} | G \sim G \\
G | \alpha, \mathcal{NW}_0 \sim DP(\alpha, \mathcal{NW}_0)\end{gathered}$$</span><br /></p>
<p>We found that standard inference algorithms like the Gibbs sampler over-estimate the number of clusters. For example, clusters containing a few samples would remain after the sampler converged, which could have been merged into a larger cluster. For this reason, we incorporated the BNPy memoized online variational inference algorithm (moVB) <span class="citation" data-cites="hughes2013memoized"></span> into our analysis framework. The moVB algorithm uses variational inference and interleaves birth, merge, and delete moves, which was shown to generate reliable clustering results by avoiding local optima and removing redundant clusters <span class="citation" data-cites="hughesBnpyReliableScalable"></span>.</p>
<h2 id="hydra-method">Hydra Method</h2>
<p>We developed a Bayesian non-parametric clustering framework for identifying biological and technical variation in large cancer gene expression datasets without the need for a reference normal dataset. To our knowledge, this is the first reproducible and widely deployable implementation of a non-parametric mixture model framework designed to overcome the challenges of precision oncology gene expression analysis. The hydra pipeline is an open source software tool hosted on GitHub (<a href="www.github.com/jpfeil/hydra">www.github.com/jpfeil/hydra</a>). A Docker container is available for deployment across environments (<a href="www.dockerhub.com/jpfeil/hydra">www.dockerhub.com/jpfeil/hydra</a>).</p>
<p>The hydra framework contains three command-line tools: filter, enrich, and sweep (Fig. <a href="#overview" data-reference-type="ref" data-reference="overview">[overview]</a>). The filter command is run first to isolate the multimodally expressed genes. Then, the user can subset these genes to pathways of interest using the enrich and sweep methods before performing multivariate clustering across genes to identify clusters of samples. The underlying analysis routines can be accessed within the docker using Jupyter notebooks. The tools to perform this analysis are contained within the docker.</p>
<figure>
<img src="img/overview-fig" alt="" /><figcaption><span><strong>Overview of the hydra framework tools.</strong></span> A: Suggested workflow for applying hydra framework tools to identify clinically relevant gene expression subtypes. B: The hydra filter command takes the large cancer gene expression matrix as input and return the multimodally expressed genes. C: The hydra enrich command takes the multimodally expressed genes and returns enriched gene-sets. The enriched gene-set genes are used for multivariate clustering of samples. D: The hydra sweep command looks for multivariate normal clusters within user-defined gene sets. This can be used for the automatic detection of clusters in large gene-set databases. <span id="overview" label="overview">[overview]</span></figcaption>
</figure>
<p>The filter command takes an expression matrix and filters down to the multimodally expressed genes using the Dirichlet Process Gaussian Mixture Model described above. This creates a directory of multimodally expressed gene models which can be used to predict differential expression in new samples. Multimodal feature selection has been shown to improve clustering performance, and the resulting clustering correlates better with clinical features <span class="citation" data-cites="yiliMultimodalityCriterionFeature2005"></span>, but there has been limited use of these methods in the pediatric precision medicine community <span class="citation" data-cites="modyIntegrativeClinicalSequencing2015 worstNextgenerationPersonalisedMedicine2016 obergImplementationNextGeneration2016"></span>.</p>
<p>The enrich and sweep routines are two independent analyses to explore multimodal expression in pediatric cancer gene expression cohorts. Both look for expression variation within a specific disease context without a matched normal comparator, which is a new analysis framework for the pediatric cancer research community. We found that multimodally expressed genes that participate in a biological pathway tend to have correlated expression distributions. This insight facilitates the detection of expression subtypes by reducing the number of genes in the clustering analysis. The hydra software comes prepackaged with popular gene sets, including the Molecular Signatures Database (MSigDB) <span class="citation" data-cites="liberzonMolecularSignaturesDatabase2011"></span>, the Gene Ontology terms <span class="citation" data-cites="ashburnerGeneOntologyTool2000 gene2018gene"></span>, and the EnrichmentMap gene sets <span class="citation" data-cites="merico2010enrichment"></span>. The gene set database is configurable, so additional gene sets can be added at runtime.</p>
<p>The enrich command uses a hypergeometric test <span class="citation" data-cites="yuClusterProfilerPackageComparing2012"></span> to discover enrichment of multimodally expressed genes within a user-defined database of gene-sets. The enrich method is used to generate new hypotheses that accelerate pediatric cancer research. The minimum component probability is an important parameter when applying the enrich command. The minimum component probability is an additional filter to remove multimodally expressed genes that influence a small subset of tumor samples. This parameter gives the user the ability subset the genes to those that influence a greater number of patients. We implemented the scan sub-routine to explore minimum component thresholds. The scan routine can be used to tune the analysis with respect to the constraints of the available data, which is an important factor in pediatric cancer research since data is often difficult to obtain. We recommend setting this threshold such that the number of genes is less than the number of samples ‚Äî the multivariate normal distribution becomes unstable when there are more genes than samples.</p>
<p>The sweep routine investigates multivariate multimodal expression signatures within a user-specific gene-set. For each gene-set, a multivariate DP-GMM is applied to determine if more than one cluster is present within the gene-set. This approach is useful when curated gene sets are available for the disease of interest, but manual inspection of each gene-set is not feasible. The sweep command can be used to identify the gene-sets with the greatest power for differentiating tumor samples. Often, gene-expression signatures are not known for a disease, which requires a more unsupervised analysis, so we recommend using the enrich command. However, the sweep command is useful when integrating known signatures that may translate to the disease of interest.</p>
<p>We have also implemented routines for cluster profiling and N-of-1 tumor analysis. These rountines are accessible within the docker container using Jupyter notebooks. Cluster profiling analysis includes GSEA to identify pathway expression that characterizes each cluster. GSEA uses all available genes since these methods require non-differentially expressed genes to assess the significance of an enrichment score. A t-statistic is calculated for each gene comparing gene expression values of samples within and outside of a cluster. Cluster profiling GSEA uses the ranked gene-level t-statistics to determine gene set enrichment.</p>
<p>The N-of-1 tumor analysis routine classifies a new gene expression profile into one of the inferred clusters, calculates a gene-level z-score for that sample relative to the normalized expression distribution, and performs GSEA. This procedure can identify new gene expression signatures that may not be detectable using the entire expression cohort as a background reference distribution. This approach is another novel contribution to the field and may facilitate the identification of clinically relative signatures that are being overlooked in current gene expression analyses.</p>
<h2 id="synthetic-data-generation-and-validation" class="unnumbered">Synthetic Data Generation and Validation</h2>
<p>We tested the hydra framework‚Äôs ability to detect differential pathway expression using synthetic cancer data. We compared hydra to two widely used gene set enrichment tools: single-sample gene set enrichment analysis (ssGSEA) and gene set variation analysis (GSVA) <span class="citation" data-cites="barbieSystematicRNAInterference2009 hanzelmannGSVAGeneSet2013 tarcaComparisonGeneSet2013"></span>. Both methods are implemented in the GSVA R package <span class="citation" data-cites="hanzelmannGSVAGeneSet2013"></span>. We modeled cancer gene expression as a multivariate Gaussian distribution. We used the TCGA glioblastoma multiforme cohort (n=166) to model a background mean and covariance matrix. This approach allowed us to model pediatric cancer gene expression data while also controlling for subtype-related expression variation. We downloaded the RSEM-quantified TPM normalized gene expression measurements from the UCSC Xena Browser <span class="citation" data-cites="goldmanUCSCXenaPlatform2018"></span>. We focus our analysis on normalized gene expression data because this data is more-widely used in the cancer research community and fewer methods are available to analysis normalized counts. To reduce heteroscedasticity and the effect of outlier expression levels, we transformed the expression data to log2(TPM + 1) <span class="citation" data-cites="zwienerTransformingRNASeqData2014"></span>.</p>
<p>We defined an expression subtype as a subset of samples with a distinct expression mean and correlation structure compared to other samples within the disease cohort. To assess the performance of the hydra approach, we generated 320 synthetic data sets each containing 300 observations. To avoid biases in the synthetic data generation process, we used random sampling to select MSigDB gene sets for each subtype, the size of the subtype <span class="math inline">Uniform(0.15,‚ÄÜ0.4)</span>, and the correlation structure within the subtype. We randomly generated the covariance matrix for the cancer subtype expression data, but used the underlying covariance matrix of the TCGA glioblastoma multiforme dataset for the background samples. We tested the effect of having 10% and 25% of genes within a gene set being differentially expressed (%DEG). In addition to these parameters, we tested a range of effect sizes: 0.25 (least different), 0.5, 0.75, 1.0, 1.5, 2.0, 2.5, and 3.0 (most different). This process was repeated twice for each gene set to create synthetic training and test data, which were then analyzed with the hydra framework using the hydra sweep command (Fig <a href="#overview" data-reference-type="ref" data-reference="overview">[overview]</a>). The mean expression filter removed any genes with a mean expression of fewer than 1.0 log2(TPM + 1). This avoids lowly-expressed genes that may have unstable expression measurements. The prior on the hydra covariance matrix was the identity scaled by 2.0, and the prior on the number of clusters was set to 2. The activated expression cluster was set to the cluster with the largest L1 norm.</p>
<h2 id="pediatric-cancer-gene-expression-data">Pediatric Cancer Gene Expression Data</h2>
<p>We downloaded pediatric cancer RNA-Seq data for neuroblastoma, osteosarcoma, Ewing sarcoma, alveolar rhabdomyosarcoma, and embryonal rhabdomyosaromca from the UCSC Treehouse Compendium (<a href="https://treehousegenomics.soe.ucsc.edu/public-data/">https://treehousegenomics.soe.ucsc.edu/public-data/</a>). This data was analyzed using the same RNA-seq pipeline, so potential computational batch effects are avoided <span class="citation" data-cites="vivianToilEnablesReproducible2017"></span>. Clinical data for the TARGET neuroblastoma and osteosarcoma samples were obtained from the TARGET Data Matrix (<a href="https://ocg.cancer.gov/programs/target/data-matrix">https://ocg.cancer.gov/programs/target/data-matrix</a>). We also analyzed a set of 58 synovial sarcoma microarray profiles with matching metastasis rate data <span class="citation" data-cites="lagardeChromosomeInstabilityAccounts2013"></span>.</p>
<h2 id="target-neuroblastoma-analysis">TARGET Neuroblastoma Analysis</h2>
<p>The neuroblastoma unsupervised enrichment analysis using the hydra enrich command included all genes with a mean expression greater than 1.0 log2(TPM + 1), a minor expression component probability of at least 20%, and a minimum effect size of 1.0. Our synthetic data analysis found that the signal decreases below an effect size of 1.0, so we use this parameter for all following analyses. We set a minimum mean threshold of 1.0 log2(TPM + 1) to exclude lowly expressed genes that may have unstable measurements. We used the hydra scan routine to search a range of minimum component probability thresholds and found 20% yielded the most number of clusters while keeping the number of genes (<span class="math inline"><em>p</em>‚ÄÑ=‚ÄÑ42</span>) below the samples (<span class="math inline"><em>n</em>‚ÄÑ=‚ÄÑ70</span>). Sample clusters were identified by applying multivariate clustering across the enriched GO term genes (FDR <span class="math inline">&lt;</span> 0.01). The multivariate mixture model <span class="math inline"><em>Œ±</em></span> concentration parameter was set to 5.0; the prior on the covariance matrix was set to the identity scaled by 2.0. The prior parameter for the number of clusters was 5. We correlated hydra expression clusters with the results of the tumor microenvironment profiling tools xCell <span class="citation" data-cites="aranXCellDigitallyPortraying2017"></span>, CIBERSORT <span class="citation" data-cites="newmanRobustEnumerationCell2015"></span>, and ESTIMATE <span class="citation" data-cites="yoshiharaInferringTumourPurity2013"></span>. We also applied the consensus clustering method M3C <span class="citation" data-cites="johnM3CMonteCarlo2018"></span> and Gap statistic k-means method <span class="citation" data-cites="tibshirani2001estimating"></span> to the TARGET MYCN-NA neuroblastoma data. We tested a range of median absolute deviation (MAD) thresholds. The number of clusters was inferred to be the smallest statistically significant value.</p>
<h2 id="small-round-blue-cell-tumor-analysis">Small Round Blue Cell Tumor Analysis</h2>
<p>We then compared the clustering patterns across small round blue cell tumors. We applied the TumorMap dimensionality reduction method <span class="citation" data-cites="newtonTumorMapExploringMolecular2017"></span> to visualize clustering of the full small round blue cell tumor gene expression matrix. We then applied the hydra framework to explore expression variation within each disease. We applied the filter/enrich approach as the TARGET MYCN-NA neuroblastoma analysis. Each disease expression matrix had unique statistical properties including sample size and level of variation. This required us to adapt the minimum probability threshold for each disease data set using the scan routine. The Jupyter notebooks for exploring these data sets can be found on GitHub (<a href="github.com/jpfeil/hydra-paper/analysis">github.com/jpfeil/hydra-paper/analysis</a>). We clustered the top 10 enriched gene sets for each disease expression cluster to show similar expression subtypes across diseases.</p>
<h2 id="statistical-analysis">Statistical Analysis</h2>
<p>A Kruskal-Wallis test was used to identify statistically significant differences across two or more groups, and a Mann-Whitney U test was used for pairwise tests using a Holm-Sidak correction for multiple hypothesis testing <span class="citation" data-cites="pedregosa2011scikit jonesSciPyOpenSource2001"></span>. We used the scipy stats implementation of the Kruskal-Wallis test and the scikit-learn post hoc processing implementation of pairwise Mann-Whitney U tests. Spearman rank and Pearson correlation values were calculated using the scipy library <span class="citation" data-cites="jonesSciPyOpenSource2001"></span>. Survival analysis was done using the survminer package <span class="citation" data-cites="kassambaraSurvminerDrawingSurvival2019"></span>.</p>
<h2 id="he-slide-preparation-and-pathologist-review">H&amp;E Slide Preparation and Pathologist Review</h2>
<p>Pediatric tumor samples were flash frozen, embedded in OCT, and 5um cryosections were collected. Slides were hematoxylin and eosin (H&amp;E) stained and imaged on a Leica DMi8, equipped with a HC PL APO 40x/0.85 NA objective and DFC7000T camera. H&amp;E slides were reviewed by a licensed pathologist for evidence of inflammation and graded as having either minimal (<span class="math inline">&lt;</span> 10%) or moderate inflammation (20-30%).</p>
<h1 id="results" class="unnumbered">Results</h1>
<h2 id="performance-assessment-using-synthetic-gene-expression-data">Performance Assessment using Synthetic Gene Expression Data</h2>
<p>To assess how well hydra detects differentially expressed pathways as compared to common GSEA approaches, we applied these methods to synthetically-generated cancer gene expression data. We generated synthetic cancer gene expression data based on the TCGA glioblastoma multiforme and the MSigDB Hallmark gene sets. We tested a range of effect sizes and percent differentially expressed genes (%DEG) within the MSigDB gene sets. We generated receiver operator curves (ROC) and calculated the Area Under the receiver operator Curve (AUC) for each analysis. Overall, the hydra pipeline outperformed the single-sample GSEA approaches with a mean AUC of 0.93 (95% CI: 0.91 - 0.95). ssGSEA had a mean AUC of 0.72 (95% CI: 0.71 - 0.74) and GSVA had a mean AUC of 0.67 (95% CI: 0.66 - 0.68) (Fig <a href="#rocplot" data-reference-type="ref" data-reference="rocplot">[rocplot]</a>A).</p>
<figure>
<img src="img/ROC-PLOT-2x" alt="" /><figcaption><span><strong>Hydra is more sensitive than available gene set enrichment approaches for detecting differential pathway expression in synthetic data.</strong></span> A: Mean receiver operator curves across effect sizes, percent differentially expressed genes (%DEG), and MSigDB Hallmark gene sets. A larger area under the curve (AUC) indicates better performance. The average AUC and 95% confidence interval for each method is in the ROC plot figure legends. B: Line plots comparing the mean AUC across a range of effect sizes and %DEG values. <span id="rocplot" label="rocplot">[rocplot]</span></figcaption>
</figure>
<p>We further investigated the performance of these methods by plotting the AUC against the effect size at %DEG of 10 and 25% (Fig <a href="#rocplot" data-reference-type="ref" data-reference="rocplot">[rocplot]</a>B). The hydra method performed better across all effect sizes, achieving near perfect performance above an effect size of 2.0 and 0.75 at %DEG of 10% and 25%, respectively. ssGSEA and GSVA performed similarly at low effect sizes, but ssGSEA performed better as the effect size increased. The ssGSEA and GSVA methods began to perform similarly to hydra at an effect size of 3.0 and %DEG of 25%. Overall, the hydra framework performed significantly better than the GSEA approaches, particularly at low effect sizes. Therefore, the hydra approach is better suited for subtyping within a disease cohort when the effect sizes are smaller and fewer genes are differentially expressed.</p>
<h2 id="hydra-analysis-of-high-risk-neuroblastoma-identifies-distinct-tumor-microenvironment-states">Hydra Analysis of High-Risk Neuroblastoma Identifies Distinct Tumor Microenvironment States</h2>
<p>After completing the performance assessment with synthetic gene expression data, we applied the hydra unsupervised enrichment analysis to the TARGET high-risk neuroblastoma cohort. High-risk neuroblastoma is an aggressive disease and is resistant to intensive therapy. Further subtyping of high-risk neuroblastoma may identify novel therapeutic targets and improve risk stratification. We hypothesized that unsupervised clustering of Gene Ontology terms would identify expression subtypes of high-risk neuroblastoma tumors. TumorMap analysis showed that the MYCN-non-amplified (MYCN-NA) neuroblastoma samples clustered separately from MYCN-amplified (MYCN-A) and stage 4S neuroblastomas samples (). We focused on the MYCN-NA neuroblastoma tumor samples because this is the largest set of samples (N=70), and variation within MYCN-NA tumors is not well understood <span class="citation" data-cites="morgensternChallengeDefiningUltrahighrisk2019a"></span>.</p>
<figure>
<img src="img/MYCN-NA-Figure-V5@2x" alt="" /><figcaption><span><strong>Hydra analysis identifies three distinct tumor microenvironment expression subtypes in MYCN non-amplified neuroblastoma samples.</strong></span> A: Gene expression heatmap displaying expression profiles of hydra clusters. Heatmap columns (samples) are ordered by hydra cluster membership. Ward hierarchical clustering applied to rows (genes) identified coordinated expression of GO term genes. B: GSEA performed on each cluster identified enrichment of tumor microenvironment and proliferative signaling gene sets. C: xCell enrichment score distributions for B-cells, CD8+ naive T-cells, and Fibroblasts, and the ESTIMATE TumorPurity score distributions for each cluster; enrichments for all cell types are available in Supplemental Tables 3-5. Abbreviations: Normalized Enrichment Score (NES), Epithelial to Mesenchymal Transition (EMT), Gene Ontology Biological Process (GOBP). <span id="MYCN-NA" label="MYCN-NA">[MYCN-NA]</span></figcaption>
</figure>
<p>We applied the hydra unsupervised enrichment analysis to the MYCN-NA cohort. The multimodal expression filter identified 428 genes with a minor component probability greater than 20% (Supplementary Table 2). Gene Ontology analysis found enrichment for the following GO terms (FDR q <span class="math inline">&lt;</span> 0.01): adaptive immune response (24 genes), mesenchyme development (12 genes), steroid hormone secretion (4 genes), and response to corticosterone (4 genes). Multivariate Dirichlet process mixture model analysis of the 44 enriched GO term genes identified three clusters of neuroblastoma samples (Fig. <a href="#MYCN-NA" data-reference-type="ref" data-reference="MYCN-NA">[MYCN-NA]</a>A). The posterior probability for belonging to each cluster was 42%, 34%, and 17%, respectively. The posterior probability for a sample belonging to a new cluster was about 6% in our analysis.</p>
<p>We next investigated cluster-specific expression signatures using GSEA (see Hydra Method section). Cluster 1 was enriched for adaptive immune response gene sets, cluster 2 was enriched for proliferative signaling gene sets, and cluster 3 was enriched for cancer-associated fibroblast gene sets (Fig <a href="#MYCN-NA" data-reference-type="ref" data-reference="MYCN-NA">[MYCN-NA]</a>B). Cluster 3 shares several features of a wound healing response, including fibroblast recruitment, extracellular matrix organization, and infiltration of immune cells <span class="citation" data-cites="fosterEvolvingRelationshipWound"></span>.</p>
<p>Clusters 1 and 3 were enriched for tumor microenvironment-associated expression. To further validate this signal, we correlated the hydra clusters with enrichment scores from the tumor microenvironment profiling tools xCell <span class="citation" data-cites="aranXCellDigitallyPortraying2017"></span> and ESTIMATE <span class="citation" data-cites="yoshiharaInferringTumourPurity2013a"></span>. Cluster 1 had higher average xCell enrichment scores associated with adaptive immune cell types including B-cells, CD4+ naive T-cells, and CD8+ naive T-cells (Kruskal-Wallis: p <span class="math inline">&lt;</span> 0.001; Supplementary Tables 3-5). Cluster 2 was characterized by the absence of immune and stromal expression and higher tumor purity. The average ESTIMATE tumor purity for each cluster was 88%, 96% and 82%, respectively. Cluster 3 was enriched for fibroblast-associated expression by xCell analysis (Kruskal-Wallis: p <span class="math inline">&lt;</span> 0.001). Clusters 1 and 3 had higher ESTIMATE immune-associated expression levels than cluster 2 (average ImmuneScore per cluster: 58, -612, 56), but cluster 3 had the highest stromal expression signature (average StromalScore per cluster: -1027, -1310, -135).</p>
<p>We next correlated clusters with clinical features. We found no difference in patient survival outcomes across clusters (log-rank test, p <span class="math inline">&gt;</span> 0.05). We note that the immune subtype in MYCN-NA neuroblastoma over-expresses cell-cycle regulation genes, which may counter-act the effect of infiltrating immune cells. We investigated associations with clinical covariates mutation burden, age, and tumor content as assessed by a clinical pathologist, but found no statistically significant differences (Kruskal-Wallis: p <span class="math inline">&gt;</span> 0.05). We then investigated associations between the hydra clusters and neuroblastoma-associated molecular aberrations and clinical features (Supplementary Table 6). ATRX gene deletions were enriched in cluster 1 (Fisher‚Äôs Exact Test: p <span class="math inline">&lt;</span> 0.05). MKI low tumors were enriched in cluster 2 and 3 (Fisher‚Äôs Exact Test: p <span class="math inline">&lt;</span> 0.01). Chromosome 17 wild type tumors were enriched in clusters 2 and 3 (Fisher‚Äôs Exact Test: p <span class="math inline">&lt;</span> 0.01).</p>
<p>Consensus clustering is a widely used approach for identifying tumor subtypes using gene expression data. We applied the M3C consensus clustering method, which is a more sophisticated version of consensus clustering that uses a null distribution to assess the statistical significance of the clustering <span class="citation" data-cites="johnM3CMonteCarlo2018 wilkersonConsensusClusterPlusClassDiscovery2010"></span>. M3C clustering of the MYCN-NA expression data using the 5000 genes with the largest median absolute deviation (MAD) resulted in the identification of two statistically significant clusters. We found that the M3C clusters correlated with the hydra clusters with lower ESTIMATE TumorPurity, but were not able to separate the adaptive immune cell and fibroblast infiltrated clusters. We also applied kmeans clustering using the gap statistic approach for estimating the number of clusters, but this approach grouped all samples into a single cluster <span class="citation" data-cites="tibshirani2001estimating maechler2012cluster"></span>. We tested a range of gene variation thresholds based on the median absolute deviation, but found similar results across thresholds (). Overall, the hydra approach was more sensitive at detecting distinct tumor microenvironment states.</p>
<h2 id="n-of-1-tumor-analysis-for-pediatric-neuroblastoma">N-of-1 Tumor Analysis for Pediatric Neuroblastoma</h2>
<p>We investigated the predictive performance of the hydra approach for identifying clinically relevant signals in the N-of-1 tumor analysis setting. We obtained tumor gene expression data from five stage 4, MYCN-NA neuroblastoma samples. The age at diagnosis ranged from 2 to 6 years. Four out of five samples had a deletion in the ATRX gene. Samples 1D and 1R are diagnosis and resection samples from the same patient. Three of the ATRX-deleted samples clustered with the high immune expression cluster (cluster 1) and one clustered in the low immune, high proliferative signaling cluster (cluster 2). Hydra analysis assigned sample 1D to cluster 1 and sample 1R to cluster 2. The resection sample 1R was extracted from lymph node tissue, which has a significantly different immune background than the training data. Another possible explanation for this change is that the tumor microenvironment is dynamic and the tumor may evade immune recognition as the disease progresses. We performed GSEA comparing samples 1D and 1R to investigate potential mechanisms leading to immune evasion in sample 1R. GSEA found downregulation of the MHC Class I Antigen Processing &amp; Presentation GO term in sample 1R (adjusted p-value <span class="math inline">&lt;</span> 0.002). Loss of antigen processing functions is a common mechanism of immune evasion across cancer types <span class="citation" data-cites="reevesAntigenProcessingImmune2017"></span>.</p>
<p>We obtained H&amp;E slide images for these samples; the images were reviewed by a licensed pathologist and scored for evidence of inflammation. The N-of-1 predictive function of the hydra toolkit was used to classify samples into the subtypes discovered by the TARGET neuroblastoma analysis. Most of the samples (4 out of 5) clustered in cluster 1, which is characterized by higher adaptive immune pathway expression. The hydra analysis agreed with the pathologist review in 4 out of 5 samples (Fig. <a href="#hefig" data-reference-type="ref" data-reference="hefig">1</a>). Sample 4 was the one discordant sample. Sample 4 is from lymph node tissue, which may have higher immune expression because of the tissue type. Notably, concordant ESTIMATE values were present in 3 out of 5 samples scored by the pathologist: samples 1D, 1R, and 2.</p>
<figure>
<img src="img/NBL-MYCN-NA-HE-2x" id="hefig" alt="" /><figcaption><span><strong>Hydra method correlates with histopathology review of tumor H&amp;E slides.</strong></span> The tumor microenvironments of stage 4, MYCN-NA neuroblastoma patient samples were analyzed using gene expression and H&amp;E slide image data. Inflammation levels in the same tumor samples were assessed from H&amp;E slide images at 20x magnification (Moderate inflammation: 20-30% lymphocyte content; Minimal inflammation: <span class="math inline">&lt;</span> 10% lymphocyte content). ATRX mutation status, hydra cluster assignment, and ESTIMATE ImmuneScore value are also indicated. Concordant and discordant predictions are marked with a positive (+) and negative (-) sign, respectively.<span label="hefig"></span></figcaption>
</figure>
<p>To further investigate expression patterns within the hydra-identified tumor microenvironment subtypes, we performed GSEA by z-score normalizing each tumor‚Äôs gene expression data to its tumor microenvironment cluster. This is a novel GSEA approach that uses the tumor microenvironment state discovered by the hydra approach to identify additional gene expression signals for individual samples. This approach revealed signals not present at the cohort level analysis (). For example, enrichment of immune expression signatures within cluster 2 predicted differences in overall survival such that patients with higher immune expression had a better overall survival rate. Similarly, an elevated cell cycle signal within cluster 3 predicted worse survival compared to other cluster 3 samples with lower cell cycle expression. This approach may provide a more appropriate background distribution for determining the significance of gene expression patterns and survival statistics.</p>
<h2 id="hydra-analysis-discovers-complex-tissue-signatures">Hydra Analysis Discovers Complex Tissue Signatures</h2>
<p>While the MYCN-NA neuroblastoma analysis focused on immune and fibroblast expression signatures, the hydra enrich method is unsupervised and is not restricted to immune or fibroblast signatures. For example, we applied the hydra filter/enrich analysis to the TARGET osteosarcoma cohort (N=74) and discovered enrichment of the GO striated muscle contraction term (FDR <span class="math inline">&lt;</span> 0.01) . Multivariate clustering for the GO striated muscle contraction gene set using the sweep routine identified two clusters. xCell analysis of the osteosarcoma cohort found significant enrichment of skeletal muscle expression in the second cluster (Mann-Whitney U test, p <span class="math inline">&lt;</span> 0.001). Surprisingly, the M3C clustering approach was not able to detect the strong muscle signature using the 5000 genes with the largest MAD (p <span class="math inline">&gt;</span> 0.05). We identified a similar expression signal in an independent cohort of osteosarcoma tumor samples and subsequently confirmed with a licensed pathologist that the tumor sample did contain muscle tissue. Explaining these sources of variation is necessary to derive clinically relevant conclusions from gene expression data. This analysis may also shed light on a biological process that is not well understood and spark a new line of research.</p>
<p>Similarly, we applied the filter method to Ewing sarcoma and discovered multimodal expression of a druggable gene, JAK1. Applying the multimodal expression model allowed us to deconstruct the Ewing sarcoma distribution into three components. We found that the subset of Ewing sarcoma tumors over-expression JAK1 also had enrichment for mast cells (. Therefore, over-expression of JAK1 may not correspond to activation of the JAK/STAT signaling pathway in cancer cells, but simply the presence of mast cells within the tumor microenvironment. Furthermore, small-molecule drugs targeting JAK1 were shown to inhibit essential mast cell functions <span class="citation" data-cites="hermansJAK1JAK2Inhibitor2018"></span>. This may be an unintended consequence of a targeted therapy approach, so methods like hydra are needed to deconstruct complex tumor expression subtypes.</p>
<h2 id="hydra-analysis-reveals-related-expression-subtypes-across-pediatric-cancers">Hydra Analysis Reveals Related Expression Subtypes Across Pediatric Cancers</h2>
<p>We next wanted to see if similar hydra clusters could be identified across small round blue cell tumors. We first performed TumorMap analysis which is a dimensionality reduction approach for visualizing genomic data on a 2D surface <span class="citation" data-cites="newtonTumorMapExploringMolecular2017"></span>. We found that MYCN-NA neuroblastoma, osteosarcoma, Ewing sarcoma, synovial sarcoma, alveolar rhabdomyosarcoma, and embyronal rhabdomyosarcoma all form distinct clusters (Fig <a href="#pancan" data-reference-type="ref" data-reference="pancan">2</a>A). This suggests there is a strong cell-of-origin signal driving the clustering of these cancer types, which is an observation that was recently made in the larger TCGA dataset <span class="citation" data-cites="Hoadley2014"></span>.</p>
<figure>
<img src="img/hydra-pan-small-round-blue-V2-2x" id="pancan" style="width:105.0%" alt="" /><figcaption><span><strong>Hydra analysis of small round blue cell tumors reveals similar expression subtypes across cancer types.</strong></span> A: TumorMap visualization of 6 small round blue cell tumor types. B: Hierarchically clustered heatmap of gene set normalized enrichment scores (NES) for the top 10 enriched gene sets for each disease type. The cancer type color bar corresponds to the cancer type color in panel A. The expression subtype was manually assigned after reviewing the most highly enriched gene sets for each cluster.<span label="pancan"></span></figcaption>
</figure>
<p>We next performed hydra analysis within each cancer type and found similar biological themes appear during the cluster profiling analysis. Hierarchical clustering of the top 10 statistically significant gene sets for each disease cluster resulted in clustering of diseases by expression subtype and not the cancer histology (Fig <a href="#pancan" data-reference-type="ref" data-reference="pancan">2</a>B). Common themes emerged across diseases including translational regulation, cell cycle regulation, immune effector cell signaling, inflammation, extracellular matrix organization, and tissue-of-origin signals. Furthermore, these signals predicted differences in patient outcomes in osteosarcoma and synovial sarcoma, two diseases for which we had patient outcome data. In both cases, the presence of immune associated expression correlated with better patient outcomes compared to tumors with proliferative signaling pathways associated with the upregulation of translation initiation.</p>
<h1 id="discussion" class="unnumbered">Discussion</h1>
<p>The hydra framework uses model-based clustering to identify recurrent expression patterns within cancer gene expression cohorts. We leveraged recent improvements in model-based clustering algorithms to identify differentially expressed genes without a matched normal distribution. We modeled differential expression as a multimodal Gaussian distribution using nonparametric Bayesian statistics. We then enriched for biologically-annotated Gene Ontology terms and performed multivariate clustering to reveal cancer subtyping expression signatures. The hydra framework can be used for identifying expression subtypes and classifying N-of-1 tumors. The hydra framework outperformed standard gene set enrichment tools for identifying overexpression of the MSigDB Hallmark cancer gene sets in synthetic cancer gene expression data and identified tumor microenvironment gene expression signatures in the TARGET pediatric neuroblastoma and osteosarcoma datasets that were not detected by consensus clustering analysis.</p>
<p>Multivariate gene expression analysis is typically underpowered because the number of genes greatly exceeds the number of samples. To address this limitation, we propose selecting for multimodally expressed genes before performing multivariate analysis. The hydra multimodal filter reduces the number of genes and enriches for genes that participate in known biological processes, including those curated in the Gene Ontology database. As Yi Li et al. (2005) found in their study on unsupervised clustering of gene expression data, we also found that reducing expression data to multimodally expressed genes improves clustering of clinical subtypes in pediatric cancers. For example, multimodally expressed genes separate neuroblastoma subtypes by TumorMap analysis better than the standard approach of using all expressed genes. Furthermore, we showed that the hydra approach is more sensitive at resolving tumor microenvironment subtypes than the M3C consensus clustering approach.</p>
<p>Significant progress has been made in subtyping neuroblastomas and adapting therapy for aggressive subtypes, but unexplained heterogeneity remains. Not accounting for this heterogeneity decreases the power of standard methods to detect important expression patterns. Identifying biomarkers using genome-wide technology may lead to improved risk stratification and the discovery of novel drug targets <span class="citation" data-cites="morgensternChallengeDefiningUltrahighrisk2019a"></span>. Hydra analysis of the TARGET MYCN-NA neuroblastoma cohort (N=70) found differential expression of tumor microenvironment markers, including markers of the adaptive immune response. Pediatric cancers are generally thought to be non-immunogenic because they have lower mutation burden than adult cancers, but the immunogenicity of pediatric cancer has not been sufficiently investigated <span class="citation" data-cites="majzner2017harnessing"></span>. Our analysis found significant variation in immune marker expression and identified ATRX deletions as a potential biomarker of immune infiltrated tumors. Further investigation into gene expression signatures and molecular aberrations that predict response to immunotherapy in pediatric cancers is needed. Hydra analysis may facilitate the development of novel therapies by grouping patients with similar tumor microenvironment properties.</p>
<p>We found significant differences in immune and stromal expression that may inform precision medicine applications. The tumor microenvironment has become an important therapeutic consideration, but few methods account for the tumor microenvironment directly. Tumor purity has been identified as a confounding factor in cancer gene expression subtyping efforts <span class="citation" data-cites="rheeImpactTumorPurity2018"></span>. For example, tumor purity and tumor microenvironment expression have been shown to correlate with pancreatic cancer subtypes <span class="citation" data-cites="raphael2017integrated"></span>. Furthermore, Aran et al. (2018) found that tumor purity was correlated with the mesenchymal glioblastoma subtype and recommended a differential expression approach to computationally remove the tumor purity signal. However, standard approaches for subtracting the tumor purity effect may not be the best approach because several mechanisms may influence tumor purity, and each mechanism may result in a different expression pattern. For instance, our analysis of MYCN-NA neuroblastoma identified two gene expression signatures that correlated with lower predicted tumor purity. Cluster 1 had an adaptive immune expression signature and cluster 3 had a cancer-associated fibroblast signature. Therefore, the estimated tumor purity signal should not be subtracted without first accounting for the different mechanisms influencing tumor purity.</p>
<p>We also found shared biological pathway enrichment across small round blue cell tumors. Although these diseases have, on the surface, distinct expression patterns, once we accounted for the cell-of-origin signal, we discovered common themes. There appears to be at least three major tumor microenvironment states: immune silent, immune infiltrated, and wound healing subtype. The immune infiltrated and wound healing subtypes predicted better overall survival in osteosarcoma and delayed metastases in synovial sarcoma tumors, which suggests the involvement of the host immune response limits the progression of tumors. Amplification of the host immune response may further limit tumor growth and lead to immune-mediated tumor cell killing.</p>
<h1 id="conclusion" class="unnumbered">Conclusion</h1>
<p>Precision oncology aims to differentiate tumors of the same diagnosis in order to match patients with the best treatment. We have developed the hydra framework to discover subtle but recurrent expression patterns within a cancer type. Our approach may help to uncover the biology underlying tumor progression and response to therapy. We have shown that hydra is more sensitive than standard gene set enrichment approaches for detecting differential pathway expression. Additionally, we applied the unsupervised hydra analysis to small round blue cell tumors and discovered distinct tumor microenvironment states. The hydra framework is a sensitive unsupervised clustering approach for N-of-1 tumor analysis and will facilitate pediatric precision oncology by discovering expression subtype signatures.</p>
<h1 id="supporting-information" class="unnumbered">Supporting information</h1>
<h4 id="s1-fig." class="unnumbered">S1 Fig.</h4>
<p><img src="img/bnpy-example-fit" alt="image" /> <span id="S1_Fig" label="S1_Fig">[S1_Fig]</span> <span><strong>Example of bnpy memoized online variational inference clustering on toy data.</strong></span> We used the bnpy moVB algorithm to infer the number of clusters from synthetic data. The model first randomly assigns clusters. Then, the model iteratively improves the model fit, creating and destroying clusters until the model converges on the correct number of clusters <span class="citation" data-cites="hughesBnpyReliableScalable"></span>.</p>
<h4 id="s2-fig." class="unnumbered">S2 Fig.</h4>
<p><img src="img/TumorMap-NBL-MM-V3-2x" alt="image" /> <span id="S2_Fig" label="S2_Fig">[S2_Fig]</span> <span><strong>Enriching for multimodally expressed genes improves clustering of established neuroblastoma subtypes.</strong></span> Standard TumorMap analysis (Newton et al. 2017) of the TARGET neuroblastoma dataset resulted in stage 4S samples clustering with stage 4 neuroblastoma samples (left). An alternative TumorMap based solely on 1,498 multimodally expressed genes separated the stage 4S samples into a distinct cluster (right).</p>
<h4 id="s3-fig." class="unnumbered">S3 Fig.</h4>
<p><img src="img/clustering-screen" style="width:75.0%" alt="image" /> <span id="S3_Fig" label="S3_Fig">[S3_Fig]</span> <span><strong><span>Consensus and k-means clustering applied to TARGET MYCN-NA dataset.</span></strong></span> We tested a range of gene expression variation thresholds based on the median absolute deviation, but found that the clusters identified by this approach could not resolve the same clusters as the hydra approach. The barplot shows the number of clusters and the lineplot tracks the Rand index comparing the M3C and k-means clusters and the hydra clusters.</p>
<h4 id="s4-fig." class="unnumbered">S4 Fig.</h4>
<p><img src="img/SubCluster-Analysis-V2@2x.png" alt="image" /> <span id="S4_Fig" label="S4_Fig">[S4_Fig]</span> <span><strong>Gene set enrichment analysis (GSEA) of MYCN-NA neuroblastoma identifies survival differences within hydra cluster 2 and cluster 3.</strong></span> A: Top 5 differentially enriched gene sets for each cluster comparing the entire MYCN-NA neuroblastoma cohort (cohort-level GSEA) and the corresponding hydra cluster (cluster-level GSEA). GSEA analysis of the hydra clusters found immune and cell cycle signals that were not identified in the cohort-level analysis. B: cluster-level GSEA separated cluster 2 into high and low immune subtypes. C: cluster-level GSEA separated cluster 3 into high and low cell cycle subtypes.</p>
<h4 id="s5-fig." class="unnumbered">S5 Fig.</h4>
<p><img src="img/muscle-signature-genes-2x" alt="image" /> <span id="S5_Fig" label="S5_Fig">[S5_Fig]</span> <span><strong>Hydra analysis of TARGET osteosarcoma cohort reveals skeletal muscle signature.</strong></span> Hydra enrichment analysis on the TARGET osteosarcoma cohort revealed a subset of patients with high skeletal muscle expression. A: Clustered heatmap shows the muscle signature genes identified by hydra unsupervised enrichment analysis. B: xCell tumor microenvironment profiling identified significant differences in skeletal muscle expression compared to background (p <span class="math inline">&lt;</span> 0.001). C: H&amp;E stained tumor slide for an independent osteosarcoma sample confirms presence of striated muscle tissue within the sequenced tumor sample.</p>
<h4 id="s6-fig." class="unnumbered">S6 Fig.</h4>
<p><img src="img/Selection_026" alt="image" /> <span id="S6_Fig" label="S6_Fig">[S6_Fig]</span> <span><strong>Hydra analysis identifies expression of druggable JAK1 gene in Ewing sarcoma correlates with mast cell expression signature.</strong></span> We show the utility of the mixture model approach for identifying important gene expression patterns for precision medicine applications. JAK1 is a druggable gene, but JAK1 is also an important gene in immune cell signaling pathways. We found that expression of JAK1 in Ewing sarcoma correlated with mast cell expression. Interestingly, the JAK1 inhibitor ruxolitinib was shown to inhibit mast cell functions <span class="citation" data-cites="hermansJAK1JAK2Inhibitor2018"></span>. Accounting for the tumor microenvironment expression is, therefore, an important step in developing precision medicine approaches for pediatric cancers.</p>
<p><span id="S1_Table" label="S1_Table">[S1_Table]</span></p>
<p><span>L<span>10cm</span>R<span>1.5cm</span>R<span>1.5cm</span></span> Pathway &amp; Adjusted P-value &amp; NES<br />
TRANSLATION (REACTOME) &amp; 1.56e-03 &amp; 4.14<br />
CHROMOSOME MAINTENANCE (REACTOME) &amp; 1.52e-03 &amp; 3.95<br />
MITOCHONDRIAL TRANSLATIONAL ELONGATION (GOBP) &amp; 1.52e-03 &amp; 3.84<br />
HALLMARK_E2F_TARGETS (MSIGDB_C2) &amp; 1.52e-03 &amp; 3.84<br />
TRANSLATION (GOBP) &amp; 1.57e-03 &amp; 3.83<br />
MITOCHONDRIAL TRANSLATION ELONGATION (REACTOME) &amp; 1.52e-03 &amp; 3.82<br />
HALLMARK_MYC_TARGETS_V1 (MSIGDB_C2) &amp; 1.52e-03 &amp; 3.80<br />
DEPOSITION OF NEW CENPA-CONTAINING NUCLEOSOMES (REACTOME) &amp; 1.52e-03 &amp; 3.78<br />
NUCLEOSOME ASSEMBLY (REACTOME) &amp; 1.52e-03 &amp; 3.78<br />
MITOCHONDRIAL TRANSLATION TERMINATION (REACTOME) &amp; 1.52e-03 &amp; 3.75<br />
</p>
<p><span>L<span>10cm</span>R<span>1.5cm</span>R<span>1.5cm</span></span> Pathway &amp; Adjusted P-value &amp; NES<br />
EXTRACELLULAR MATRIX ORGANIZATION (GOBP) &amp; 0.04 &amp; 4.38<br />
EXTRACELLULAR STRUCTURE ORGANIZATION (GOBP) &amp; 0.05 &amp; 4.38<br />
NABA_CORE_MATRISOME (MSIGDB_C2) &amp; 0.04 &amp; 4.02<br />
HALLMARK_INFLAMMATORY_RESPONSE (MSIGDB_C2) &amp; 0.03 &amp; 3.97<br />
TYROBP CAUSAL NETWORK (WIKIPATHWAYS_20181110) &amp; 0.01 &amp; 3.93<br />
PID_AVB3_INTEGRIN_PATHWAY (MSIGDB_C2) &amp; 0.02 &amp; 3.91<br />
HALLMARK_COAGULATION (MSIGDB_C2) &amp; 0.02 &amp; 3.90<br />
HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION (MSIGDB_C2) &amp; 0.04 &amp; 3.89<br />
PID_INTEGRIN1_PATHWAY (MSIGDB_C2) &amp; 0.01 &amp; 3.88<br />
BETA1 INTEGRIN CELL SURFACE INTERACTIONS (PID) &amp; 0.01 &amp; 3.88<br />
</p>
<p><img src="img/osteo-kaplan-meier-2x" style="width:75.0%" alt="image" /></p>
<p><span id="S2_Table" label="S2_Table">[S2_Table]</span></p>
<p><span>L<span>10cm</span>R<span>1.5cm</span>R<span>1.5cm</span></span> Pathway &amp; Adjusted P-Value &amp; NES<br />
TRANSLATION (REACTOME) &amp; 3.36e-03 &amp; 4.52<br />
TRANSLATION (GOBP) &amp; 3.36e-03 &amp; 4.42<br />
RIBOSOME BIOGENESIS (GOBP) &amp; 3.36e-03 &amp; 4.42<br />
RNA PROCESSING (REACTOME) &amp; 3.36e-03 &amp; 4.29<br />
RRNA PROCESSING IN THE NUCLEUS AND CYTOSOL (REACTOME) &amp; 3.36e-03 &amp; 4.25<br />
MAJOR PATHWAY OF RRNA PROCESSING (REACTOME) &amp; 3.36e-03 &amp; 4.22<br />
PEPTIDE BIOSYNTHETIC PROCESS (GOBP) &amp; 3.36e-03 &amp; 4.17<br />
RRNA PROCESSING (GOBP) &amp; 3.36e-03 &amp; 4.16<br />
NCRNA PROCESSING (GOBP) &amp; 3.36e-03 &amp; 4.16<br />
HALLMARK_MYC_TARGETS_V1 (MSIGDB_C2) &amp; 3.36e-03 &amp; 4.09<br />
</p>
<p><span>L<span>10cm</span>R<span>1.5cm</span>R<span>1.5cm</span></span> Pathway &amp; Adjusted P-value &amp; NES<br />
HALLMARK_INTERFERON_GAMMA_RESPONSE (MSIGDB_C2) &amp; 1.77e-03 &amp; 5.11<br />
HALLMARK_INFLAMMATORY_RESPONSE (MSIGDB_C2) &amp; 1.77e-03 &amp; 5.01<br />
HALLMARK_TNFA_SIGNALING_VIA_NFKB (MSIGDB_C2) &amp; 1.77e-03 &amp; 4.87<br />
RESPONSE TO INTERFERON-GAMMA (GOBP) &amp; 1.77e-03 &amp; 4.77<br />
INFLAMMATORY RESPONSE (GOBP) &amp; 1.77e-03 &amp; 4.67<br />
LEUKOCYTE MIGRATION (GOBP) &amp; 1.77e-03 &amp; 4.63<br />
HALLMARK_ALLOGRAFT_REJECTION (MSIGDB_C2) &amp; 1.77e-03 &amp; 4.60<br />
CELLULAR RESPONSE TO INTERFERON-GAMMA (GOBP) &amp; 1.77e-03 &amp; 4.57<br />
IMMUNOREGULATORY INTERACTIONS &amp; 1.77e-03 &amp; 4.57<br />
MYELOID CELL ACTIVATION (REACTOME) &amp; 1.77e-03 &amp; 4.56<br />
</p>
<p><img src="img/synovial-kaplan-meier-2x" style="width:75.0%" alt="image" /></p>
<h1 id="acknowledgments" class="unnumbered">Acknowledgments</h1>
<p>We would like to thank the patients and families who participated in translational genomics research. We would like to thank St. Baldrick‚Äôs Foundation, California Precision Medicine Initiative, National Human Genome Research Institute of the National Institutes of Health for funding support.</p>
